From 6e4228cb86633dc0f02f2b765ea47b8fbf0cafdf Mon Sep 17 00:00:00 2001
From: ahs <amrita.h.s@intel.com>
Date: Tue, 17 Mar 2020 18:26:39 +0530
Subject: Use the new PRODUCT_SUPPORTS_VERITY variable

Tracked-On:
Signed-off-by: ahs <amrita.h.s@intel.com>
---
 Android.mk | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/Android.mk b/Android.mk
index b797424..96e81e6 100644
--- a/Android.mk
+++ b/Android.mk
@@ -185,7 +185,8 @@ LOCAL_MODULE := kernelflinger-$(TARGET_BUILD_VARIANT)
 
 
 # if dm-verity is disabled for eng purpose skip the oem-cert
-ifeq ($(PRODUCTS.$(INTERNAL_PRODUCT).PRODUCT_SUPPORTS_VERITY), true)
+#ifeq ($(PRODUCTS.$(INTERNAL_PRODUCT).PRODUCT_SUPPORTS_VERITY), true)
+ifeq ($(PRODUCT_SUPPORTS_VERITY), true)
 kf_intermediates := $(call intermediates-dir-for,EFI,kernelflinger)
 
 VERITY_CERT := $(kf_intermediates)/verity.cer
@@ -284,7 +285,8 @@ endif
 
 LOCAL_CFLAGS := $(SHARED_CFLAGS)
 
-ifeq ($(PRODUCTS.$(INTERNAL_PRODUCT).PRODUCT_SUPPORTS_VERITY), true)
+#ifeq ($(PRODUCTS.$(INTERNAL_PRODUCT).PRODUCT_SUPPORTS_VERITY), true)
+ifeq ($(PRODUCT_SUPPORTS_VERITY), true)
 LOCAL_OBJCOPY_FLAGS := -j .oemkeys
 endif
 
@@ -423,7 +425,8 @@ ifneq ($(strip $(KERNELFLINGER_USE_UI)),false)
     LOCAL_SRC_FILES += \
         ux.c
 endif
-ifeq ($(PRODUCTS.$(INTERNAL_PRODUCT).PRODUCT_SUPPORTS_VERITY),true)
+#ifeq ($(PRODUCTS.$(INTERNAL_PRODUCT).PRODUCT_SUPPORTS_VERITY),true)
+ifeq ($(PRODUCT_SUPPORTS_VERITY),true)
 keys4abl_intermediates := $(call intermediates-dir-for,ABL,keys)
 
 ABL_VERITY_CERT := $(keys4abl_intermediates)/verity.cer
-- 
2.24.0

