From f4b840f6d50920f41227180c47f0475b7c9f9b76 Mon Sep 17 00:00:00 2001
From: "Ai, Ting" <ting.a.ai@intel.com>
Date: Wed, 24 Aug 2022 10:19:06 +0800
Subject: [PATCH] Enforce device to unlock status

Signed-off-by: Ai, Ting <ting.a.ai@intel.com>
---
 kernelflinger.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/kernelflinger.c b/kernelflinger.c
index 3ecba2a..e05f7b9 100644
--- a/kernelflinger.c
+++ b/kernelflinger.c
@@ -1251,8 +1251,7 @@ EFI_STATUS efi_main(EFI_HANDLE image, EFI_SYSTEM_TABLE *sys_table)
 	/* For civ, flash images to disk is not MUST. So set device to LOCKED
 	 * state by default on the first boot.
 	*/
-	if (need_lock)
-		set_current_state(LOCKED);
+	set_current_state(UNLOCKED);
 
 	ret = set_device_security_info(NULL);
 	if (EFI_ERROR(ret)) {
-- 
2.39.2

