From 597a12b61a9fdea072dc2e52cadbd163e6fc6738 Mon Sep 17 00:00:00 2001
From: rnaidu <ramya.v.naidu@intel.com>
Date: Tue, 31 Mar 2020 07:50:25 +0530
Subject: [PATCH] Fixing the allowzip64 issue for r2 celadon images

Tracked-On: N/A
Signed-off-by: rnaidu <ramya.v.naidu@intel.com>
---
 Lib/zipfile.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Lib/zipfile.py b/Lib/zipfile.py
index 3ab66cea69..ba90ee3063 100644
--- a/Lib/zipfile.py
+++ b/Lib/zipfile.py
@@ -723,7 +723,7 @@ class ZipFile(object):
 
     fp = None                   # Set here since __del__ checks it
 
-    def __init__(self, file, mode="r", compression=ZIP_STORED, allowZip64=False):
+    def __init__(self, file, mode="r", compression=ZIP_STORED, allowZip64=True):
         """Open the ZIP file with mode read "r", write "w" or append "a"."""
         if mode not in ("r", "w", "a"):
             raise RuntimeError('ZipFile() requires mode "r", "w", or "a"')
-- 
2.24.1

