From b0440b17ebea0531b45250a9ae7395044aa0eb0a Mon Sep 17 00:00:00 2001
From: devath1x <devathix.mahesh@intel.com>
Date: Tue, 1 Oct 2024 12:12:01 +0000
Subject: [PATCH] [Do not merge] wifi coverity

Signed-off-by: devath1x <devathix.mahesh@intel.com>
---
 libwifi_hal/hal_tool.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libwifi_hal/hal_tool.cpp b/libwifi_hal/hal_tool.cpp
index 49652554c..eb1eaba9e 100644
--- a/libwifi_hal/hal_tool.cpp
+++ b/libwifi_hal/hal_tool.cpp
@@ -411,7 +411,7 @@ wifi_error wifi_nan_beacon_sdf_payload_request_stub(
 }
 
 wifi_error wifi_nan_register_handler_stub(wifi_interface_handle iface,
-                                          NanCallbackHandler handlers) {
+                                          NanCallbackHandler* handlers) {
   return WIFI_ERROR_NOT_SUPPORTED;
 }
 
-- 
2.34.1

