From f6646ddc2c9d662e8d9bb4417f032c9fd5ef4cad Mon Sep 17 00:00:00 2001
From: revatipx <revatix.prasad@intel.com>
Date: Wed, 4 Sep 2019 15:01:48 +0530
Subject: [PATCH] test ptp mtp switch issue Tracked-On: OAM85336

---
 drivers/usb/dwc3/core.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/dwc3/core.h b/drivers/usb/dwc3/core.h
index 13102850..bba6486 100644
--- a/drivers/usb/dwc3/core.h
+++ b/drivers/usb/dwc3/core.h
@@ -32,7 +32,7 @@
 #define DWC3_MSG_MAX	500
 
 /* Global constants */
-#define DWC3_PULL_UP_TIMEOUT	500	/* ms */
+#define DWC3_PULL_UP_TIMEOUT	1000	/* ms */
 #define DWC3_BOUNCE_SIZE	1024	/* size of a superspeed bulk */
 #define DWC3_EP0_SETUP_SIZE	512
 #define DWC3_ENDPOINTS_NUM	32
-- 
1.9.1

