From ea6bfcf4e454380d743b225f69908c35a7b07a6b Mon Sep 17 00:00:00 2001
From: "Wan, Hao" <haox.wan@intel.com>
Date: Thu, 11 Apr 2024 15:40:48 +0000
Subject: [PATCH] Enable mediasdk sw decoder in caas_cvmb

Signed-off-by: Wan, Hao <haox.wan@intel.com>
---
 caas_cvmb/mixins.spec | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/caas_cvmb/mixins.spec b/caas_cvmb/mixins.spec
index 920b3c2..0c5144a 100755
--- a/caas_cvmb/mixins.spec
+++ b/caas_cvmb/mixins.spec
@@ -25,15 +25,15 @@ cpu-arch: x86
 allow-missing-dependencies: true
 dexpreopt: true
 pstore: false
-media: auto(enable_msdk_omx=false, add_sw_msdk=false, opensource_msdk=true, opensource_msdk_omx_il=false)
+media: auto(enable_msdk_omx=true, add_sw_msdk=true, add_hw_msdk=false, opensource_msdk=false, opensource_msdk_omx_il=false)
 graphics: auto(gen9+=true,vulkan=true,minigbm=true,gralloc1=true,enable_guc=false)
 storage: sdcard-v-usb-only(adoptablesd=false,adoptableusb=false)
 ethernet: dhcp
 camera-ext: ext-camera-only
 rfkill: true(force_disable=)
 wlan: iwlwifi(libwifi-hal=true)
-codecs: configurable(sw_omx_video=false, hw_omx_video=false, platform=tgl, profile_file=media_profiles_1080p.xml, gpu=gen12)
-codec2: true(enable_msdk_c2=true, use_onevpl=true, platform=adl, hw_ve_vp9=true, hw_vd_vp8=false)
+codecs: configurable(sw_omx_video=false, hw_omx_video=false, platform=tgl, profile_file=media_profiles_1080p.xml, gpu=gen12, sw_vd_h264=true, sw_vd_h265=true)
+codec2: android-c2
 usb: host
 usb-gadget: auto(usb_config=adb,mtp_adb_pid=0x0a5f,ptp_adb_pid=0x0a61,rndis_pid=0x0a62,rndis_adb_pid=0x0a63,bcdDevice=0x0,bcdUSB=0x200,controller=dwc3.2.auto,f_acm=false,f_dvc_trace=true,dvctrace_source_dev=dvcith-0-msc0)
 midi: true
-- 
2.34.1

