From 525427e5c972f6a853f7421c3bffc4ba52153266 Mon Sep 17 00:00:00 2001
From: celadon <celadon@intel.com>
Date: Thu, 4 Aug 2022 05:17:43 +0000
Subject: [PATCH] virtual box expects sampling freq 44.1k

---
 primary/audio_hw.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/primary/audio_hw.c b/primary/audio_hw.c
index e4394c5..6e022d5 100755
--- a/primary/audio_hw.c
+++ b/primary/audio_hw.c
@@ -52,12 +52,12 @@
 
 #define OUT_PERIOD_SIZE 1024
 #define OUT_PERIOD_COUNT 4
-#define OUT_SAMPLING_RATE 48000
+#define OUT_SAMPLING_RATE 44100
 
 #define IN_PERIOD_SIZE 1024 //default period size
 #define IN_PERIOD_MS 10
 #define IN_PERIOD_COUNT 4
-#define IN_SAMPLING_RATE 48000
+#define IN_SAMPLING_RATE 44100
 
 #define AUDIO_PARAMETER_HFP_ENABLE   "hfp_enable"
 #define AUDIO_PARAMETER_BT_SCO       "BT_SCO"
-- 
2.33.1

