From dff3a87539605dd33b11f6e48f432a32b72eb77e Mon Sep 17 00:00:00 2001
From: "Chen, Gang G" <gang.g.chen@intel.com>
Date: Thu, 15 Jun 2023 21:54:40 +0800
Subject: [PATCH 06/54] Solve no enough memory for string copy issue

Tracked-On: OAM-110767
Signed-off-by: Chen, Gang G <gang.g.chen@intel.com>
---
 libfastboot/fastboot.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libfastboot/fastboot.c b/libfastboot/fastboot.c
index 605ed98..f57924b 100644
--- a/libfastboot/fastboot.c
+++ b/libfastboot/fastboot.c
@@ -63,7 +63,7 @@
 /* size of "INFO" "OKAY" or "FAIL" */
 #define CODE_LENGTH 4
 #define INFO_PAYLOAD (MAGIC_LENGTH - CODE_LENGTH)
-#define MAX_VARIABLE_LENGTH 64
+#define MAX_VARIABLE_LENGTH 256
 #if defined(IOC_USE_SLCAN) || defined(IOC_USE_CBC)
 #define TIMEOUT 5
 #endif
-- 
2.25.1

