From 942f55d82273f9f40b1f91341f2d4fce84380e04 Mon Sep 17 00:00:00 2001
From: HeYue <yue.he@intel.com>
Date: Fri, 11 Feb 2022 13:02:05 +0800
Subject: [PATCH] WA to debug the qemu 6.0 boot fail

Signed-off-by: HeYue <yue.he@intel.com>
---
 arch/x86/kernel/time.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/x86/kernel/time.c b/arch/x86/kernel/time.c
index e42faa792c07..57f3a3d674df 100644
--- a/arch/x86/kernel/time.c
+++ b/arch/x86/kernel/time.c
@@ -100,8 +100,8 @@ static __init void x86_late_time_init(void)
 	x86_init.irqs.intr_mode_init();
 	tsc_init();
 
-	if (static_cpu_has(X86_FEATURE_WAITPKG))
-		use_tpause_delay();
+//	if (static_cpu_has(X86_FEATURE_WAITPKG))
+//		use_tpause_delay();
 }
 
 /*
-- 
2.17.1

