From 06243f5b4ac651276101af9893eb1cdeded409e0 Mon Sep 17 00:00:00 2001
From: jizhenlo <zhenlong.z.ji@intel.com>
Date: Fri, 22 Jul 2022 16:33:13 +0800
Subject: [PATCH] Disable the ratelimit for print

Signed-off-by: jizhenlo <zhenlong.z.ji@intel.com>
---
 include/linux/ratelimit_types.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/linux/ratelimit_types.h b/include/linux/ratelimit_types.h
index b676aa419eef..fe576802bb3a 100644
--- a/include/linux/ratelimit_types.h
+++ b/include/linux/ratelimit_types.h
@@ -6,8 +6,8 @@
 #include <linux/param.h>
 #include <linux/spinlock_types.h>
 
-#define DEFAULT_RATELIMIT_INTERVAL	(5 * HZ)
-#define DEFAULT_RATELIMIT_BURST		10
+#define DEFAULT_RATELIMIT_INTERVAL	(0 * HZ)
+#define DEFAULT_RATELIMIT_BURST		1000000
 
 /* issue num suppressed message on exit */
 #define RATELIMIT_MSG_ON_RELEASE	BIT(0)
-- 
2.25.1

