From 7a912156d2965219bdb30fb76ff589c1ac5f2e91 Mon Sep 17 00:00:00 2001
From: "Babu B, Ramesh" <ramesh.babu.b@intel.com>
Date: Wed, 5 Aug 2020 20:49:03 +0530
Subject: [PATCH] EventHub check for class cursor

Tracked-On:
Signed-off-by: Babu B, Ramesh <ramesh.babu.b@intel.com>
---
 services/inputflinger/EventHub.cpp | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/services/inputflinger/EventHub.cpp b/services/inputflinger/EventHub.cpp
index ce5627271..1592c373a 100644
--- a/services/inputflinger/EventHub.cpp
+++ b/services/inputflinger/EventHub.cpp
@@ -1285,6 +1285,13 @@ status_t EventHub::openDeviceLocked(const char* devicePath) {
     ioctl(fd, EVIOCGBIT(EV_FF, sizeof(device->ffBitmask)), device->ffBitmask);
     ioctl(fd, EVIOCGPROP(sizeof(device->propBitmask)), device->propBitmask);
 
+    ALOGV("HyperV :: keyBitmask = %s \n", device->keyBitmask);
+    ALOGV("HyperV :: relBitmask = %s \n", device->relBitmask);
+    ALOGV("HyperV :: swBitmask = %s \n", device->swBitmask);
+    ALOGV("HyperV :: ledBitmask = %s \n", device->ledBitmask);
+    ALOGV("HyperV :: ffBitmask = %s \n", device->ffBitmask);
+    ALOGV("HyperV :: propBitmask = %s \n", device->propBitmask);
+
     // See if this is a keyboard.  Ignore everything in the button range except for
     // joystick and gamepad buttons which are handled like keyboards for the most part.
     bool haveKeyboardKeys = containsNonZeroByte(device->keyBitmask, 0, sizeof_bit_array(BTN_MISC))
@@ -1299,12 +1306,18 @@ status_t EventHub::openDeviceLocked(const char* devicePath) {
     }
 
     // See if this is a cursor device such as a trackball or mouse.
+    //if (test_bit(BTN_MOUSE, device->keyBitmask)
     if (test_bit(BTN_MOUSE, device->keyBitmask)
             && test_bit(REL_X, device->relBitmask)
             && test_bit(REL_Y, device->relBitmask)) {
         device->classes |= INPUT_DEVICE_CLASS_CURSOR;
     }
 
+    if((identifier.vendor == 0x045E) && (identifier.product == 0x0621)) {
+	ALOGV("HyperV :: INPUT_DEVICE_CLASS_CURSOR \n");
+	device->classes |= INPUT_DEVICE_CLASS_CURSOR;
+    }
+
     // See if this is a rotary encoder type device.
     String8 deviceType = String8();
     if (device->configuration &&
-- 
2.27.0

