From 25aefbb14d520d716f2366c4fde0443582edf011 Mon Sep 17 00:00:00 2001
From: "Babu B, Ramesh" <ramesh.babu.b@intel.com>
Date: Thu, 6 Aug 2020 01:33:35 +0530
Subject: [PATCH] Mouse event check

Tracked-On:
Signed-off-by: Babu B, Ramesh <ramesh.babu.b@intel.com>
---
 services/inputflinger/EventHub.cpp | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/services/inputflinger/EventHub.cpp b/services/inputflinger/EventHub.cpp
index ce5627271..8777deb31 100644
--- a/services/inputflinger/EventHub.cpp
+++ b/services/inputflinger/EventHub.cpp
@@ -1285,6 +1285,13 @@ status_t EventHub::openDeviceLocked(const char* devicePath) {
     ioctl(fd, EVIOCGBIT(EV_FF, sizeof(device->ffBitmask)), device->ffBitmask);
     ioctl(fd, EVIOCGPROP(sizeof(device->propBitmask)), device->propBitmask);
 
+    ALOGV("HyperV :: keyBitmask = %s \n", device->keyBitmask);
+    ALOGV("HyperV :: relBitmask = %s \n", device->relBitmask);
+    ALOGV("HyperV :: swBitmask = %s \n", device->swBitmask);
+    ALOGV("HyperV :: ledBitmask = %s \n", device->ledBitmask);
+    ALOGV("HyperV :: ffBitmask = %s \n", device->ffBitmask);
+    ALOGV("HyperV :: propBitmask = %s \n", device->propBitmask);
+
     // See if this is a keyboard.  Ignore everything in the button range except for
     // joystick and gamepad buttons which are handled like keyboards for the most part.
     bool haveKeyboardKeys = containsNonZeroByte(device->keyBitmask, 0, sizeof_bit_array(BTN_MISC))
@@ -1299,12 +1306,26 @@ status_t EventHub::openDeviceLocked(const char* devicePath) {
     }
 
     // See if this is a cursor device such as a trackball or mouse.
+    //if (test_bit(BTN_MOUSE, device->keyBitmask)
     if (test_bit(BTN_MOUSE, device->keyBitmask)
             && test_bit(REL_X, device->relBitmask)
             && test_bit(REL_Y, device->relBitmask)) {
+	ALOGV("HyperV :: INPUT_DEVICE_CLASS_CURSOR REL_XY \n");
+        device->classes |= INPUT_DEVICE_CLASS_CURSOR;
+    } else if (test_bit(BTN_MOUSE, device->keyBitmask)
+               && test_bit(ABS_X, device->absBitmask)
+               && test_bit(ABS_Y, device->absBitmask)
+               && test_bit(REL_WHEEL, device->relBitmask)) {
+	ALOGV("HyperV :: INPUT_DEVICE_CLASS_CURSOR ABS_XY \n");
         device->classes |= INPUT_DEVICE_CLASS_CURSOR;
     }
 
+/*
+    if((identifier.vendor == 0x045E) && (identifier.product == 0x0621)) {
+	ALOGV("HyperV :: INPUT_DEVICE_CLASS_CURSOR \n");
+	device->classes |= INPUT_DEVICE_CLASS_CURSOR;
+    }
+*/
     // See if this is a rotary encoder type device.
     String8 deviceType = String8();
     if (device->configuration &&
-- 
2.27.0

