From 058b2056f2a14ff9511e73c453f419be51ba925c Mon Sep 17 00:00:00 2001
From: "Liu, Kai1" <kai1.liu@intel.com>
Date: Wed, 20 Mar 2024 11:22:29 +0800
Subject: [PATCH] let ns accessiable

---
 linker/linker.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/linker/linker.cpp b/linker/linker.cpp
index 712e2bb04..b18eb7964 100644
--- a/linker/linker.cpp
+++ b/linker/linker.cpp
@@ -1210,6 +1210,7 @@ static bool load_library(android_namespace_t* ns,
 
   // do not check accessibility using realpath if fd is located on tmpfs
   // this enables use of memfd_create() for apps
+/*
   if ((fs_stat.f_type != TMPFS_MAGIC) && (!ns->is_accessible(realpath))) {
     // TODO(dimitry): workaround for http://b/26394120 - the exempt-list
 
@@ -1253,6 +1254,7 @@ static bool load_library(android_namespace_t* ns,
       return false;
     }
   }
+*/
 
   soinfo* si = soinfo_alloc(ns, realpath.c_str(), &file_stat, file_offset, rtld_flags);
 
-- 
2.33.1

