From 6d1b353d90256c4983ba846ebe4e1aae0c8f5a42 Mon Sep 17 00:00:00 2001
From: "Cui, Yuxin" <yuxin.cui@intel.com>
Date: Fri, 25 Feb 2022 15:24:28 +0800
Subject: [PATCH] Add checkDumpPermission for soundtrigger_middleware

Tracked-On: OAM-100531
Signed-off-by: Cui, Yuxin <yuxin.cui@intel.com>
---
 .../SoundTriggerMiddlewareService.java                   | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/services/core/java/com/android/server/soundtrigger_middleware/SoundTriggerMiddlewareService.java b/services/core/java/com/android/server/soundtrigger_middleware/SoundTriggerMiddlewareService.java
index 4b464d2d3e04..d21cc8fbfe1e 100644
--- a/services/core/java/com/android/server/soundtrigger_middleware/SoundTriggerMiddlewareService.java
+++ b/services/core/java/com/android/server/soundtrigger_middleware/SoundTriggerMiddlewareService.java
@@ -31,6 +31,7 @@ import android.os.RemoteException;
 import android.util.Log;
 
 import com.android.server.SystemService;
+import com.android.internal.util.DumpUtils;
 
 import java.io.FileDescriptor;
 import java.io.PrintWriter;
@@ -64,13 +65,15 @@ public class SoundTriggerMiddlewareService extends ISoundTriggerMiddlewareServic
 
     @NonNull
     private final ISoundTriggerMiddlewareInternal mDelegate;
+    private final Context mContext;
 
     /**
      * Constructor for internal use only. Could be exposed for testing purposes in the future.
      * Users should access this class via {@link Lifecycle}.
      */
-    private SoundTriggerMiddlewareService(@NonNull ISoundTriggerMiddlewareInternal delegate) {
+    private SoundTriggerMiddlewareService(@NonNull ISoundTriggerMiddlewareInternal delegate, @NonNull Context context) {
         mDelegate = Objects.requireNonNull(delegate);
+        mContext = context;
         new ExternalCaptureStateTracker(active -> {
             try {
                 mDelegate.setCaptureState(active);
@@ -94,6 +97,7 @@ public class SoundTriggerMiddlewareService extends ISoundTriggerMiddlewareServic
     }
 
     @Override protected void dump(FileDescriptor fd, PrintWriter fout, String[] args) {
+        if (!DumpUtils.checkDumpPermission(mContext, TAG, fout)) return;
         if (mDelegate instanceof Dumpable) {
             ((Dumpable) mDelegate).dump(fout);
         }
@@ -185,7 +189,8 @@ public class SoundTriggerMiddlewareService extends ISoundTriggerMiddlewareServic
                                     new SoundTriggerMiddlewareValidation(
                                             new SoundTriggerMiddlewareImpl(factories,
                                                     new AudioSessionProviderImpl()),
-                                            getContext()))));
+                                            getContext())),
+                            getContext()));
         }
     }
 }
-- 
2.25.1

