From b2995c743a6f64ba1501ceded1c0870c9af3df49 Mon Sep 17 00:00:00 2001
From: rramakax <rohitx.kant@intel.com>
Date: Fri, 14 Aug 2020 13:19:14 +0530
Subject: [PATCH] Given Fix for Null pointer check and added default argument
 to the getboolean function

Tracked-On:16011496842
Signed-off-by: rramakax <rohitx.kant@intel.com>

diff --git a/src/com/android/settings/applications/manageapplications/ManageApplications.java b/src/com/android/settings/applications/manageapplications/ManageApplications.java
index 971c6f5..d547a42 100644
--- a/src/com/android/settings/applications/manageapplications/ManageApplications.java
+++ b/src/com/android/settings/applications/manageapplications/ManageApplications.java
@@ -318,7 +318,7 @@ public class ManageApplications extends InstrumentedFragment
             mShowSystem = savedInstanceState.getBoolean(EXTRA_SHOW_SYSTEM, mShowSystem);
             mFilterType =
                     savedInstanceState.getInt(EXTRA_FILTER_TYPE, AppFilterRegistry.FILTER_APPS_ALL);
-            mExpandSearch = savedInstanceState.getBoolean(EXTRA_EXPAND_SEARCH_VIEW);
+            mExpandSearch = savedInstanceState.getBoolean(EXTRA_EXPAND_SEARCH_VIEW, false);
         }
 
         mInvalidSizeStr = activity.getText(R.string.invalid_size_value);
@@ -508,7 +508,9 @@ public class ManageApplications extends InstrumentedFragment
         outState.putBoolean(EXTRA_SHOW_SYSTEM, mShowSystem);
         outState.putBoolean(EXTRA_HAS_ENTRIES, mApplications.mHasReceivedLoadEntries);
         outState.putBoolean(EXTRA_HAS_BRIDGE, mApplications.mHasReceivedBridgeCallback);
+        if (mSearchView != null) {
         outState.putBoolean(EXTRA_EXPAND_SEARCH_VIEW, !mSearchView.isIconified());
+        }
         outState.putInt(EXTRA_FILTER_TYPE, mFilter.getFilterType());
         if (mApplications != null) {
             mApplications.onSaveInstanceState(outState);
-- 
1.9.1

