From 75e73c273f4ff9e3b76ebdd4b89e8feffec16407 Mon Sep 17 00:00:00 2001
From: HeYue <yue.he@intel.com>
Date: Wed, 8 Mar 2023 15:06:14 +0800
Subject: [PATCH] Add two macro for Android S build

Tracked-On: OAM-106720
Signed-off-by: HeYue <yue.he@intel.com>
---
 meson.build | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/meson.build b/meson.build
index eca34ea21a89..15ac19f320f9 100644
--- a/meson.build
+++ b/meson.build
@@ -1605,6 +1605,8 @@ endif
 if with_platform_android
   pre_args += '-DHAVE_TIMESPEC_GET'
   pre_args += '-DHAVE_LINUX_FUTEX_H'
+  pre_args += '-DHAVE_STRUCT_TIMESPEC'
+  pre_args += '-DHAVE_FUNC_ATTRIBUTE_PACKED'
 endif
 
 # only used in Iris and ANV
-- 
2.17.1

