From 3b93031b80ba4b811a3a771e72d067b086864be7 Mon Sep 17 00:00:00 2001
From: "Lin, Shenghua" <shenghua.lin@intel.com>
Date: Thu, 11 Apr 2024 02:44:31 +0000
Subject: [PATCH] Disable ray tracing

---
 meson.build | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/meson.build b/meson.build
index de18e8272d0..25bd79ca738 100644
--- a/meson.build
+++ b/meson.build
@@ -297,14 +297,12 @@ with_any_broadcom = [
 
 if ['x86_64'].contains(host_machine.cpu_family())
   with_intel_clc = get_option('intel-clc') == 'enabled'
+  with_intel_vk_rt = with_intel_vk and get_option('intel-clc') != 'disabled'
 else
   with_intel_clc = false
+  with_intel_vk_rt = false
 endif
 
-# Enable ray tracing without checking intel-clc since we have compiled opencl c shaders on linux and
-# copy them to android code.
-with_intel_vk_rt = with_intel_vk
-
 with_any_intel = [
   with_gallium_crocus,
   with_gallium_i915,
-- 
2.34.1

