From 119fc450a783dcbcede559e1ebb441510300af19 Mon Sep 17 00:00:00 2001
From: anuvarsh <anuvarshini.bc@intel.com>
Date: Fri, 1 Feb 2019 15:25:23 +0530
Subject: [PATCH] Adding -march=haswell in libart

Tracked-On:
---
 build/Android.bp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/build/Android.bp b/build/Android.bp
index 2a5598f..f2da623 100644
--- a/build/Android.bp
+++ b/build/Android.bp
@@ -35,7 +35,6 @@ art_global_defaults {
         "-Wshadow",
         "-Wunused",
         "-fvisibility=protected",
-
         // Warn about thread safety violations with clang.
         "-Wthread-safety",
         "-Wthread-safety-negative",
@@ -63,6 +62,7 @@ art_global_defaults {
         "-D_LIBCPP_ENABLE_THREAD_SAFETY_ANNOTATIONS",
     ],
 
+
     target: {
         android: {
             cflags: [
@@ -73,6 +73,7 @@ art_global_defaults {
                 // "-fno-omit-frame-pointer",
                 // "-marm",
                 // "-mapcs",
+                   "-march=haswell",
             ],
             include_dirs: [
                 // We optimize Thread::Current() with a direct TLS access. This requires access to a
-- 
1.9.1

