From 5d2b268ebc07937ff9a6626e6ca26ea73754e2c3 Mon Sep 17 00:00:00 2001
From: svenate <salini.venate@intel.com>
Date: Fri, 2 Jul 2021 14:54:10 +0530
Subject: [PATCH] Logs in verifier

Signed-off-by: svenate <salini.venate@intel.com>
---
 runtime/parsed_options.cc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/runtime/parsed_options.cc b/runtime/parsed_options.cc
index c62caa98de..2bb533e345 100644
--- a/runtime/parsed_options.cc
+++ b/runtime/parsed_options.cc
@@ -503,7 +503,7 @@ bool ParsedOptions::ProcessSpecialOptions(const RuntimeOptions& options,
 
 // Intended for local changes only.
 static void MaybeOverrideVerbosity() {
-  //  gLogVerbosity.class_linker = true;  // TODO: don't check this in!
+    gLogVerbosity.class_linker = true;  // TODO: don't check this in!
   //  gLogVerbosity.collector = true;  // TODO: don't check this in!
   //  gLogVerbosity.compiler = true;  // TODO: don't check this in!
   //  gLogVerbosity.deopt = true;  // TODO: don't check this in!
@@ -521,7 +521,7 @@ static void MaybeOverrideVerbosity() {
   //  gLogVerbosity.startup = true;  // TODO: don't check this in!
   //  gLogVerbosity.third_party_jni = true;  // TODO: don't check this in!
   //  gLogVerbosity.threads = true;  // TODO: don't check this in!
-  //  gLogVerbosity.verifier = true;  // TODO: don't check this in!
+    gLogVerbosity.verifier = true;  // TODO: don't check this in!
 }
 
 bool ParsedOptions::DoParse(const RuntimeOptions& options,
-- 
2.17.1

