From e3b926fb64b4fa426e0002331bd2421d793380e3 Mon Sep 17 00:00:00 2001
From: Priyanka Bose <priyanka.bose@intel.com>
Date: Fri, 23 Oct 2020 19:04:54 +0530
Subject: [PATCH] ART-Runtime-PGO-optimizations

Signed-off-by: Priyanka Bose <priyanka.bose@intel.com>
---
 runtime/Android.bp | 52 ++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 52 insertions(+)

diff --git a/runtime/Android.bp b/runtime/Android.bp
index c9ab61c6c5..2cd2c98e43 100644
--- a/runtime/Android.bp
+++ b/runtime/Android.bp
@@ -23,6 +23,49 @@ JIT_DEBUG_REGISTER_CODE_LDFLAGS = [
     "-Wl,--keep-unique,__dex_debug_register_code",
 ]
 
+
+libart_cc_defaults {
+     name: "libart-pgo-defaults",
+     pgo: {
+         instrumentation: true,
+         benchmarks: ["art_runtime", "dex2oat",],
+     },
+     target: {
+         android_arm64: {
+             pgo: {
+                 profile_file: "art/art-runtime_arm_arm64.profdata",
+             },
+         },
+         android_arm: {
+             pgo: {
+                 profile_file: "art/art-runtime_arm_arm64.profdata",
+             },
+         },
+         android_x86_64: {
+             pgo: {
+                 profile_file: "art/libart-x86-x86_64.profdata.2019-06-06",
+             },
+         },
+         android_x86: {
+             pgo: {
+                 profile_file: "art/libart-x86-x86_64.profdata.2019-06-06",
+             },
+         },
+	 android_mips64: {
+             pgo: {
+                 profile_file: "art/art-runtime_mips_mips64.profdata",
+             },
+         },
+         android_mips: {
+             pgo: {
+                 profile_file: "art/art-runtime_mips_mips64.profdata",
+             },
+         },
+     },
+}
+
+
+
 // These are defaults for native shared libaries that are expected to be
 // in stack traces often.
 libart_cc_defaults {
@@ -524,6 +567,7 @@ art_cc_library {
     defaults: [
         "libart_defaults",
         "libart_nativeunwind_defaults",
+	"libart-pgo-defaults",
     ],
     whole_static_libs: [
     ],
@@ -540,6 +584,14 @@ art_cc_library {
     export_shared_lib_headers: [
         "libdexfile",
     ],
+    pgo: {
+         // Additional cflags just for dex2oat during PGO instrumentation
+         cflags: [
+               // Ignore frame-size increase resulting from instrumentation.
+               "-Wno-frame-larger-than=",
+               "-DART_PGO_INSTRUMENTATION",
+         ],
+    },
     target: {
         android: {
             lto: {
-- 
2.28.0

