From 48dec5a92d950c8dc9e7442430e65f189559465d Mon Sep 17 00:00:00 2001
From: ahs <amrita.h.s@intel.com>
Date: Tue, 8 Dec 2020 12:40:03 +0530
Subject: [PATCH] Patch avx versions of libart and libartd with correct soname

Tracked-On:
Signed-off-by: ahs <amrita.h.s@intel.com>
---
 runtime/Android.bp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/runtime/Android.bp b/runtime/Android.bp
index ab9ea08..993c289 100644
--- a/runtime/Android.bp
+++ b/runtime/Android.bp
@@ -578,6 +578,7 @@ art_cc_library {
             asflags: ["-DMTERP_USE_AVX"],
         },
     },
+    ldflags: ["-Wl,-soname,libart.so"],
 }
 
 art_cc_library {
@@ -598,6 +599,7 @@ art_cc_library {
             asflags: ["-DMTERP_USE_AVX"],
         },
     },
+    ldflags: ["-Wl,-soname,libartd.so"],
 }
 
 libart_cc_defaults {
-- 
2.7.4

