From 018f8a19ca2bf161908a379c13d7180b537a8c32 Mon Sep 17 00:00:00 2001
From: "Chen, Gang G" <gang.g.chen@intel.com>
Date: Thu, 16 May 2024 09:28:35 +0800
Subject: [PATCH 54/54] Fix compile error which avbtool not found

Tracked-On: OAM-118715
Signed-off-by: Chen, Gang G <gang.g.chen@intel.com>
---
 Android.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Android.mk b/Android.mk
index a832373..e1e54f0 100644
--- a/Android.mk
+++ b/Android.mk
@@ -387,7 +387,7 @@ BOOTLOADER_AVB_KEY_PATH := $(BOARD_AVB_KEY_PATH)
 endif
 
 $(ABL_AVB_PK): $(BOOTLOADER_AVB_KEY_PATH)
-	external/avb/avbtool extract_public_key --key $< --output $@
+	external/avb/avbtool.py extract_public_key --key $< --output $@
 
 $(ABL_PADDED_AVB_PK): $(ABL_AVB_PK)
 	$(call pad-binary, 4096)
-- 
2.25.1

