From d619788194d482ceb6aff35ab9e9b41dca893bd5 Mon Sep 17 00:00:00 2001
From: sgnanase <sundar.gnanasekaran@intel.com>
Date: Tue, 7 Nov 2023 12:11:48 +0530
Subject: [PATCH] Ensure we override caas_dev to caas

Signed-off-by: sgnanase <sundar.gnanasekaran@intel.com>
---
 core/Makefile   | 1 +
 core/sysprop.mk | 9 ++++++++-
 2 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/core/Makefile b/core/Makefile
index 80abd1936b..bcc7d70e5d 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -3211,6 +3211,7 @@ define build-systemimage-target
   @mkdir -p $(dir $(1)) $(systemimage_intermediates) && rm -rf $(systemimage_intermediates)/system_image_info.txt
   $(call generate-image-prop-dictionary, $(systemimage_intermediates)/system_image_info.txt,system, \
       skip_fsck=true)
+  $(call override-targetname-property)
   PATH=$(INTERNAL_USERIMAGES_BINARY_PATHS):$$PATH \
       $(BUILD_IMAGE) \
           $(TARGET_OUT) $(systemimage_intermediates)/system_image_info.txt $(1) $(TARGET_OUT) \
diff --git a/core/sysprop.mk b/core/sysprop.mk
index bd6f3d9fdf..fee52fa66b 100644
--- a/core/sysprop.mk
+++ b/core/sysprop.mk
@@ -558,4 +558,11 @@ ALL_INSTALLED_BUILD_PROP_FILES := \
 # $1 installed file path, e.g. out/target/product/vsoc_x86_64/system/build.prop
 define is-build-prop
 $(if $(findstring $1,$(ALL_INSTALLED_BUILD_PROP_FILES)),Y)
-endef
\ No newline at end of file
+endef
+
+define override-targetname-property
+  @echo "XOXOXO: Override caas_dev to caas"
+  cd $(PRODUCT_OUT)/system && sed -i 's/caas_dev/caas/g' build.prop && cd -
+endef
+
+$(call override-targetname-property)
-- 
2.40.1

