From bc5f465399c3d3f007e422c14022148b88f66bac Mon Sep 17 00:00:00 2001
From: "Shi, Qiming" <qiming.shi@intel.com>
Date: Fri, 2 Aug 2024 05:06:13 +0000
Subject: [PATCH] Make c2 service 64bit

Signed-off-by: Shi, Qiming <qiming.shi@intel.com>
---
 c2_store/Android.bp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/c2_store/Android.bp b/c2_store/Android.bp
index aca3302..0b0032b 100644
--- a/c2_store/Android.bp
+++ b/c2_store/Android.bp
@@ -61,7 +61,6 @@ cc_binary {
     ],
     required: ["android.hardware.media.c2-vendor-seccomp_policy"],
 
-    compile_multilib: "32",
     // The content in manifest_media_c2_default.xml can be included
     // directly in the main device manifest.xml file or via vintf_fragments.
     // (Remove the line below if the entry is already in the main manifest.)
-- 
2.34.1

