From 92c267cb68772ad90abba41a8617acf14a74ff46 Mon Sep 17 00:00:00 2001
From: "Shi, Qiming" <qiming.shi@intel.com>
Date: Fri, 2 Aug 2024 05:06:13 +0000
Subject: [PATCH] Make c2 service 64bit

Signed-off-by: Shi, Qiming <qiming.shi@intel.com>
---
 c2_store/Android.bp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/c2_store/Android.bp b/c2_store/Android.bp
index 64c044e..9926ea9 100644
--- a/c2_store/Android.bp
+++ b/c2_store/Android.bp
@@ -63,7 +63,6 @@ cc_binary {
 
     required: ["android.hardware.media.c2@1.0-vendor.policy"],
 
-    compile_multilib: "32",
 }
 
 prebuilt_etc {
-- 
2.34.1

