From 177fe8411d6fa5a78429f278d884cee6944b535c Mon Sep 17 00:00:00 2001
From: anuvarsh <anuvarshini.bc@intel.com>
Date: Fri, 16 Aug 2019 12:06:12 +0530
Subject: [PATCH] script-api Android.bp

Tracked-On:
Signed-off-by: anuvarsh <anuvarshini.bc@intel.com>
---
 script_api/Android.bp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/script_api/Android.bp b/script_api/Android.bp
index 8f42e475..cd972255 100644
--- a/script_api/Android.bp
+++ b/script_api/Android.bp
@@ -17,6 +17,8 @@ cc_binary_host {
     cflags: [
         "-Wall",
         "-Werror",
+	"-mavx2",
+	"-mfma",
     ],
 
     sanitize: {
-- 
2.21.0

