From f162f90b21a4935986aadbe4f3647404a074d4be Mon Sep 17 00:00:00 2001
From: Salini Venate <salini.venate@intel.com>
Date: Wed, 20 Mar 2024 09:54:29 +0530
Subject: [PATCH] test min sdk version

Signed-off-by: Salini Venate <salini.venate@intel.com>
---
 java/android_manifest.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/java/android_manifest.go b/java/android_manifest.go
index f2ebfa6a2..4e13e4aab 100644
--- a/java/android_manifest.go
+++ b/java/android_manifest.go
@@ -106,7 +106,7 @@ func ManifestFixer(ctx android.ModuleContext, manifest android.Path,
 		if err != nil {
 			ctx.ModuleErrorf("invalid minSdkVersion: %s", err)
 		}
-		if minSdkVersion.FinalOrFutureInt() >= 23 {
+		if minSdkVersion.FinalOrFutureInt() >= 20 {
 			args = append(args, fmt.Sprintf("--extract-native-libs=%v", !params.UseEmbeddedNativeLibs))
 		} else if params.UseEmbeddedNativeLibs {
 			ctx.ModuleErrorf("module attempted to store uncompressed native libraries, but minSdkVersion=%s doesn't support it",
-- 
2.17.1

