From 5091c8f3b0b912919895d8e6dd188f407b1ec77f Mon Sep 17 00:00:00 2001
From: ahs <amrita.h.s@intel.com>
Date: Mon, 21 Nov 2022 10:11:35 +0000
Subject: [PATCH] Enable avx and avx2 flag for full android build

Signed-off-by: ahs <amrita.h.s@intel.com>
---
 cc/config/x86_64_device.go | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/cc/config/x86_64_device.go b/cc/config/x86_64_device.go
index e4c3c7fe2..4011961cc 100644
--- a/cc/config/x86_64_device.go
+++ b/cc/config/x86_64_device.go
@@ -73,8 +73,8 @@ var (
 		// flags so these flags are not enabled by default.
 		// if there is performance gain in individual library components,
 		// the compiler flags can be set in corresponding bp files.
-		// "avx":    []string{"-mavx"},
-		// "avx2":   []string{"-mavx2"},
+		 "avx":    []string{"-mavx"},
+		 "avx2":   []string{"-mavx2"},
 		// "avx512": []string{"-mavx512"}
 
 		"popcnt": []string{"-mpopcnt"},
-- 
2.37.1

