From dadeb8e67915a9baff00b94d8730a9aa282c11d6 Mon Sep 17 00:00:00 2001
From: anuvarsh <anuvarshini.bc@intel.com>
Date: Fri, 16 Aug 2019 11:53:53 +0530
Subject: [PATCH] cpp rs Android.bp

Tracked-On:
Signed-off-by: anuvarsh <anuvarshini.bc@intel.com>
---
 cpp/Android.bp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/cpp/Android.bp b/cpp/Android.bp
index fabb9cbc..21929884 100644
--- a/cpp/Android.bp
+++ b/cpp/Android.bp
@@ -15,6 +15,8 @@ cc_library_static {
         "-Wno-unused-parameter",
         "-std=c++11",
         "-DRS_COMPATIBILITY_LIB",
+	"-mavx2",
+	"-mfma",
     ],
 
 
@@ -53,6 +55,8 @@ cc_defaults {
         "-Wextra",
         "-Wno-unused-parameter",
         "-Wno-unused-variable",
+	"-mavx2",
+	"-mfma",
     ],
 
     // We need to export not just rs/cpp but also rs.  This is because
@@ -84,7 +88,7 @@ cc_library_static {
     name: "libRScpp_static",
     defaults: ["libRScpp-defaults"],
 
-    cflags: ["-DRS_COMPATIBILITY_LIB"],
+    cflags: ["-DRS_COMPATIBILITY_LIB", "-mavx2", "-mfma"],
 
     sdk_version: "9",
     whole_static_libs: ["libRSDispatch"],
-- 
2.21.0

