From a1e378ef50abfdb6dad0c38d2323ec7d25e82c96 Mon Sep 17 00:00:00 2001
From: swaroopb <swaroop.balan@intel.com>
Date: Tue, 24 Sep 2019 09:46:49 +0530
Subject: [PATCH] [CELADON] VtsHalGraphicsMapperV2_1Target module is skipped

Tracked-On: OAM-86798
Signed-off-by: swaroopb <swaroop.balan@intel.com>
---
 vendor/file_contexts | 1 +
 1 file changed, 1 insertion(+)

diff --git a/vendor/file_contexts b/vendor/file_contexts
index 9da79f463..1fd8a4078 100644
--- a/vendor/file_contexts
+++ b/vendor/file_contexts
@@ -69,6 +69,7 @@
 # Same process HALs installed by platform into /vendor
 #
 /(vendor|system/vendor)/lib(64)?/hw/android\.hardware\.graphics\.mapper@2\.0-impl\.so u:object_r:same_process_hal_file:s0
+/(vendor|system/vendor)/lib(64)?/hw/android\.hardware\.graphics\.mapper@2\.0-impl-2\.1\.so u:object_r:same_process_hal_file:s0
 /(vendor|system/vendor)/lib(64)?/hw/android\.hardware\.graphics\.mapper@3\.0-impl\.so u:object_r:same_process_hal_file:s0
 /(vendor|system/vendor)/lib(64)?/hw/android\.hardware\.renderscript@1\.0-impl\.so     u:object_r:same_process_hal_file:s0
 /(vendor|system/vendor)/lib(64)?/hw/gralloc\.default\.so                              u:object_r:same_process_hal_file:s0
-- 
2.22.0

