From 6ae153e099ac9716f5bb1309af723235207bf845 Mon Sep 17 00:00:00 2001
From: Raveendra Babu Chennakesavulu <raveendra.babu.chennakesavulu@intel.com>
Date: Wed, 20 May 2020 23:09:35 +0530
Subject: [PATCH] [Framework] ethernet proxy changes

Change-Id: I82669e623f853026d2abd9736b759b79872e0535
Tracked-On:

diff --git a/java/com/android/server/ethernet/EthernetNetworkFactory.java b/java/com/android/server/ethernet/EthernetNetworkFactory.java
index de0a691..7eee0d0 100644
--- a/java/com/android/server/ethernet/EthernetNetworkFactory.java
+++ b/java/com/android/server/ethernet/EthernetNetworkFactory.java
@@ -269,6 +269,8 @@ public class EthernetNetworkFactory extends NetworkFactory {
         void setIpConfig(IpConfiguration ipConfig) {
 
             this.mIpConfig = ipConfig;
+	    updateLinkState(false);
+	    updateLinkState(true);
         }
 
         boolean statisified(NetworkCapabilities requestedCapabilities) {
-- 
2.17.1

