From 82ef1de14949e47043e360ced2a87d38467d7c1b Mon Sep 17 00:00:00 2001
From: kanlihu <kanli.hu@intel.com>
Date: Sat, 19 Mar 2022 03:41:43 +0800
Subject: [PATCH] only enable layer 1 disable 0,2,3,4,5

---
 DrmHwcTwo.cpp       | 2 +-
 backend/Backend.cpp | 3 ++-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/DrmHwcTwo.cpp b/DrmHwcTwo.cpp
index db4c528..96f11ff 100644
--- a/DrmHwcTwo.cpp
+++ b/DrmHwcTwo.cpp
@@ -644,7 +644,7 @@ HWC2::Error DrmHwcTwo::HwcDisplay::CreateComposition(bool test) {
   for (std::pair<const hwc2_layer_t, DrmHwcTwo::HwcLayer> &l : layers_) {
     switch (l.second.validated_type()) {
       case HWC2::Composition::Device:
-        z_map.emplace(std::make_pair(l.second.z_order(), &l.second));
+        //z_map.emplace(std::make_pair(l.second.z_order(), &l.second));
         break;
       case HWC2::Composition::Client:
         // Place it at the z_order of the lowest client layer
diff --git a/backend/Backend.cpp b/backend/Backend.cpp
index 44afd14..d87b968 100644
--- a/backend/Backend.cpp
+++ b/backend/Backend.cpp
@@ -65,6 +65,7 @@ HWC2::Error Backend::ValidateDisplay(DrmHwcTwo::HwcDisplay *display,
   return *num_types ? HWC2::Error::HasChanges : HWC2::Error::None;
 }
 
+const std::set<int> enable_set{0,1};
 std::tuple<int, size_t> Backend::GetClientLayers(
     DrmHwcTwo::HwcDisplay *display,
     const std::vector<DrmHwcTwo::HwcLayer *> &layers) {
@@ -112,7 +113,7 @@ uint32_t Backend::CalcPixOps(const std::vector<DrmHwcTwo::HwcLayer *> &layers,
 void Backend::MarkValidated(std::vector<DrmHwcTwo::HwcLayer *> &layers,
                             size_t client_first_z, size_t client_size) {
   for (int z_order = 0; z_order < layers.size(); ++z_order) {
-    if (z_order >= client_first_z && z_order < client_first_z + client_size)
+    if (z_order == 1 && z_order >= client_first_z && z_order < client_first_z + client_size)
       layers[z_order]->set_validated_type(HWC2::Composition::Client);
     else
       layers[z_order]->set_validated_type(HWC2::Composition::Device);
-- 
2.31.0

