From 86c3986a9e037597d7d5c4b82723f3dd697ad077 Mon Sep 17 00:00:00 2001
From: "Liu, Kai1" <kai1.liu@intel.com>
Date: Mon, 10 Jul 2023 14:21:40 +0800
Subject: [PATCH] release_to_os_ms=5000

---
 standalone/allocator_config.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/standalone/allocator_config.h b/standalone/allocator_config.h
index 8e103f2..d9eb5e7 100644
--- a/standalone/allocator_config.h
+++ b/standalone/allocator_config.h
@@ -94,16 +94,16 @@ struct AndroidConfig {
   static const uptr PrimaryRegionSizeLog = 18U;
   typedef uptr PrimaryCompactPtrT;
 #endif
-  static const s32 PrimaryMinReleaseToOsIntervalMs = 1000;
-  static const s32 PrimaryMaxReleaseToOsIntervalMs = 1000;
+  static const s32 PrimaryMinReleaseToOsIntervalMs = 5000;
+  static const s32 PrimaryMaxReleaseToOsIntervalMs = 5000;
 
   typedef MapAllocatorCache<AndroidConfig> SecondaryCache;
   static const u32 SecondaryCacheEntriesArraySize = 256U;
   static const u32 SecondaryCacheQuarantineSize = 32U;
   static const u32 SecondaryCacheDefaultMaxEntriesCount = 32U;
   static const uptr SecondaryCacheDefaultMaxEntrySize = 2UL << 20;
-  static const s32 SecondaryCacheMinReleaseToOsIntervalMs = 0;
-  static const s32 SecondaryCacheMaxReleaseToOsIntervalMs = 1000;
+  static const s32 SecondaryCacheMinReleaseToOsIntervalMs = 5000;
+  static const s32 SecondaryCacheMaxReleaseToOsIntervalMs = 5000;
 
   template <class A>
   using TSDRegistryT = TSDRegistrySharedT<A, 8U, 2U>; // Shared, max 8 TSDs.
-- 
2.33.1

