From b0e84f1202ca17ee6c22ed588ee188bf8ec41b6c Mon Sep 17 00:00:00 2001
From: "Liu, Kai1" <kai1.liu@intel.com>
Date: Tue, 11 Jul 2023 15:11:18 +0800
Subject: [PATCH] add tache len to 20

---
 standalone/size_class_map.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/standalone/size_class_map.h b/standalone/size_class_map.h
index 1948802..1b566a2 100644
--- a/standalone/size_class_map.h
+++ b/standalone/size_class_map.h
@@ -254,7 +254,7 @@ struct AndroidSizeClassConfig {
   static const uptr MinSizeLog = 4;
   static const uptr MidSizeLog = 7;
   static const uptr MaxSizeLog = 16;
-  static const u32 MaxNumCachedHint = 14;
+  static const u32 MaxNumCachedHint = 20;
   static const uptr MaxBytesCachedLog = 13;
 
   static constexpr u32 Classes[] = {
-- 
2.33.1

