From e8a84019c1d52567a1494fb81af094df0ff48288 Mon Sep 17 00:00:00 2001
From: "Liu, Kai1" <kai1.liu@intel.com>
Date: Thu, 13 Jul 2023 09:27:39 +0800
Subject: [PATCH] scudo: enlarge secondaryCacheEntry and secondaryCacheMaxSize

---
 standalone/allocator_config.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/standalone/allocator_config.h b/standalone/allocator_config.h
index 8e103f2..957b02a 100644
--- a/standalone/allocator_config.h
+++ b/standalone/allocator_config.h
@@ -100,10 +100,10 @@ struct AndroidConfig {
   typedef MapAllocatorCache<AndroidConfig> SecondaryCache;
   static const u32 SecondaryCacheEntriesArraySize = 256U;
   static const u32 SecondaryCacheQuarantineSize = 32U;
-  static const u32 SecondaryCacheDefaultMaxEntriesCount = 32U;
-  static const uptr SecondaryCacheDefaultMaxEntrySize = 2UL << 20;
-  static const s32 SecondaryCacheMinReleaseToOsIntervalMs = 0;
-  static const s32 SecondaryCacheMaxReleaseToOsIntervalMs = 1000;
+  static const u32 SecondaryCacheDefaultMaxEntriesCount = 256U;
+  static const uptr SecondaryCacheDefaultMaxEntrySize = 40UL << 20;
+  static const s32 SecondaryCacheMinReleaseToOsIntervalMs = 5000;
+  static const s32 SecondaryCacheMaxReleaseToOsIntervalMs = 5000;
 
   template <class A>
   using TSDRegistryT = TSDRegistrySharedT<A, 8U, 2U>; // Shared, max 8 TSDs.
-- 
2.33.1

