From b16cdb778adbd9d8f6fabf88bf630aad39bbcf91 Mon Sep 17 00:00:00 2001
From: "Liu, Kai1" <kai1.liu@intel.com>
Date: Mon, 10 Jul 2023 14:34:00 +0800
Subject: [PATCH] modify secondary cache

---
 standalone/allocator_config.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/standalone/allocator_config.h b/standalone/allocator_config.h
index 8e103f2..d300e22 100644
--- a/standalone/allocator_config.h
+++ b/standalone/allocator_config.h
@@ -98,10 +98,10 @@ struct AndroidConfig {
   static const s32 PrimaryMaxReleaseToOsIntervalMs = 1000;
 
   typedef MapAllocatorCache<AndroidConfig> SecondaryCache;
-  static const u32 SecondaryCacheEntriesArraySize = 256U;
-  static const u32 SecondaryCacheQuarantineSize = 32U;
-  static const u32 SecondaryCacheDefaultMaxEntriesCount = 32U;
-  static const uptr SecondaryCacheDefaultMaxEntrySize = 2UL << 20;
+  static const u32 SecondaryCacheEntriesArraySize = 1024U;
+  static const u32 SecondaryCacheQuarantineSize = 8U;
+  static const u32 SecondaryCacheDefaultMaxEntriesCount = 128U;
+  static const uptr SecondaryCacheDefaultMaxEntrySize = 8UL << 20;
   static const s32 SecondaryCacheMinReleaseToOsIntervalMs = 0;
   static const s32 SecondaryCacheMaxReleaseToOsIntervalMs = 1000;
 
-- 
2.33.1

