From 33e9c5bba4cacadc0c10b1ba7205d75f78133826 Mon Sep 17 00:00:00 2001
From: "Liu, Kai1" <kai1.liu@intel.com>
Date: Thu, 6 Jul 2023 14:10:37 +0800
Subject: [PATCH] change primary size to 256k in scudo

---
 standalone/size_class_map.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/standalone/size_class_map.h b/standalone/size_class_map.h
index 1948802..f1d6380 100644
--- a/standalone/size_class_map.h
+++ b/standalone/size_class_map.h
@@ -238,7 +238,7 @@ struct AndroidSizeClassConfig {
   static const uptr NumBits = 7;
   static const uptr MinSizeLog = 4;
   static const uptr MidSizeLog = 6;
-  static const uptr MaxSizeLog = 16;
+  static const uptr MaxSizeLog = 18;
   static const u32 MaxNumCachedHint = 13;
   static const uptr MaxBytesCachedLog = 13;
 
@@ -246,7 +246,7 @@ struct AndroidSizeClassConfig {
       0x00020, 0x00030, 0x00040, 0x00050, 0x00060, 0x00070, 0x00090, 0x000b0,
       0x000c0, 0x000e0, 0x00120, 0x00160, 0x001c0, 0x00250, 0x00320, 0x00450,
       0x00670, 0x00830, 0x00a10, 0x00c30, 0x01010, 0x01210, 0x01bd0, 0x02210,
-      0x02d90, 0x03790, 0x04010, 0x04810, 0x05a10, 0x07310, 0x08210, 0x10010,
+      0x02d90, 0x03790, 0x04010, 0x04810, 0x05a10, 0x07310, 0x08210, 0x40010,
   };
   static const uptr SizeDelta = 16;
 #else
-- 
2.33.1

