From b7227220ba841d028c22d30672d1dca5b534da7f Mon Sep 17 00:00:00 2001
From: "Liu, Kai1" <kai1.liu@intel.com>
Date: Mon, 10 Jul 2023 13:12:12 +0800
Subject: [PATCH] add quarantine flag

---
 standalone/flags.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/standalone/flags.cpp b/standalone/flags.cpp
index de5153b..d5c1430 100644
--- a/standalone/flags.cpp
+++ b/standalone/flags.cpp
@@ -57,7 +57,7 @@ static const char *getCompileDefinitionScudoDefaultOptions() {
 }
 
 static const char *getScudoDefaultOptions() {
-  return (&__scudo_default_options) ? __scudo_default_options() : "";
+  return (&__scudo_default_options) ? __scudo_default_options() : "quarantine_size_kb=512:thread_local_quarantine_size_kb=256:quarantine_max_chunk_size=1024";
 }
 
 void initFlags() {
-- 
2.33.1

