From 38b8f790ea5090fcd02c417421bbd5f5ff2a4d5d Mon Sep 17 00:00:00 2001
From: "Liu, Kai1" <kai1.liu@intel.com>
Date: Tue, 11 Jul 2023 15:04:06 +0800
Subject: [PATCH] support primary release_to_os_ms

---
 standalone/primary64.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/standalone/primary64.h b/standalone/primary64.h
index 94375fc..4c0df52 100644
--- a/standalone/primary64.h
+++ b/standalone/primary64.h
@@ -185,7 +185,7 @@ public:
         continue;
       RegionInfo *Region = getRegionInfo(I);
       ScopedLock L(Region->Mutex);
-      TotalReleasedBytes += releaseToOSMaybe(Region, I, /*Force=*/true);
+      TotalReleasedBytes += releaseToOSMaybe(Region, I, /*Force=*/false);
     }
     return TotalReleasedBytes;
   }
-- 
2.33.1

