From 3f382c02e0e2c09377d838093cd0a65af14a11e1 Mon Sep 17 00:00:00 2001
From: bodapati <shalini.salomi.bodapati@intel.com>
Date: Thu, 16 Jul 2020 15:29:12 +0530
Subject: [PATCH] Enable avx and fma flags in libart module

Tracked-On:
Signed-off-by: bodapati <shalini.salomi.bodapati@intel.com>
---
 build/Android.bp | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/build/Android.bp b/build/Android.bp
index 7b807d56b7..c50cd90f33 100644
--- a/build/Android.bp
+++ b/build/Android.bp
@@ -107,6 +107,24 @@ art_global_defaults {
         // Enable thread annotations for std::mutex, etc.
         "-D_LIBCPP_ENABLE_THREAD_SAFETY_ANNOTATIONS",
     ],
+    arch: {
+        x86: {
+            avx2: {
+                cflags: [
+                    "-mavx2",
+                    "-mfma",
+                ],
+            },
+        },
+        x86_64: {
+            avx2: {
+                cflags: [
+                    "-mavx2",
+                    "-mfma",
+                ],
+            },
+        },
+    },
 
     target: {
         android: {
-- 
2.17.1

