From 08a2a88421cb76505ee3ba4d5c271041729335c1 Mon Sep 17 00:00:00 2001
From: Ratnesh Kumar Rai <ratnesh.kumar.rai@intel.com>
Date: Fri, 14 Jul 2023 21:45:27 +0530
Subject: [PATCH] Fixed crashes due to wrong environment name

Signed-off-by: Ratnesh Kumar Rai <ratnesh.kumar.rai@intel.com>
---
 va/Android.mk | 2 ++
 va/va.c       | 2 +-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/va/Android.mk b/va/Android.mk
index 75334db..5359364 100644
--- a/va/Android.mk
+++ b/va/Android.mk
@@ -58,6 +58,7 @@ LOCAL_CFLAGS := \
 	$(IGNORED_WARNNING) \
         -DLOG_TAG=\"libva\" \
         -DSYSCONFDIR="\"$(LIBVA_CONFIG_DIR)\""
+        -DANDROID=1 \
 
 LOCAL_C_INCLUDES := \
 	$(LOCAL_PATH)/..
@@ -123,6 +124,7 @@ LOCAL_SRC_FILES := \
 
 LOCAL_CFLAGS += \
 	-DLOG_TAG=\"libva-android\" \
+        -DANDROID=1 \
 	$(IGNORED_WARNNING)
 
 LOCAL_C_INCLUDES += \
diff --git a/va/va.c b/va/va.c
index fb62442..25f1d7c 100644
--- a/va/va.c
+++ b/va/va.c
@@ -60,7 +60,7 @@
 #ifndef HAVE_SECURE_GETENV
 static char * secure_getenv(const char *name)
 {
-#if defined(__MINGW32__) || defined(__MINGW64__)
+#if defined(__MINGW32__) || defined(__MINGW64__) || defined(ANDROID)
     if (getuid() == geteuid())
 #else
     if (getuid() == geteuid() && getgid() == getegid())
-- 
2.17.1

