From 2bcad1d53237e2ce4fd251748174d7b1da67ba72 Mon Sep 17 00:00:00 2001
From: Sapna <sapna1.singh@intel.com>
Date: Thu, 3 Apr 2025 10:49:00 +0530
Subject: [PATCH] Make the headers available for other projects

Signed-off-by: Sapna <sapna1.singh@intel.com>
---
 Android.bp | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/Android.bp b/Android.bp
index 162c230b6..63e49672f 100644
--- a/Android.bp
+++ b/Android.bp
@@ -48,6 +48,15 @@ license {
     ],
 }
 
+cc_library_headers {
+    name: "freetype_headers",
+
+    export_include_dirs: [
+        "include",
+        "src/gzip",
+    ],
+}
+
 cc_defaults {
     name: "libft2_defaults",
     host_supported: true,
@@ -89,6 +98,7 @@ cc_defaults {
         "src/svg/svg.c",
         "src/truetype/truetype.c",
         "src/type1/type1.c",
+        "src/base/ftsynth.c",
     ],
 
     export_include_dirs: ["include"],
-- 
2.42.0

