From bbf82f60eee086972b3f28e1cc54cb13ff14977d Mon Sep 17 00:00:00 2001
From: gkdeepa <g.k.deepa@intel.com>
Date: Tue, 30 Nov 2021 16:08:25 +0530
Subject: [PATCH] error handling during preview

Tracked-On:
---
 src/CameraSocketServerThread.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/CameraSocketServerThread.cpp b/src/CameraSocketServerThread.cpp
index 285c621..ad23064 100644
--- a/src/CameraSocketServerThread.cpp
+++ b/src/CameraSocketServerThread.cpp
@@ -663,6 +663,11 @@ bool CameraSocketServerThread::threadLoop() {
                      if(trans_mode == VSOCK){
                         while(size_update != 460800){
                             size = recv(mClientFd, (char *)fbuffer+size_update, 460800, 0);
+                            if(size < 0){
+                               //error handling while in preview
+                               ALOGE(LOG_TAG "entered into recv error, break to recover");
+                               break;
+                            }
                             size_update += size;
                             if (size_update == 460800){
                                 handle->clientRevCount++;
-- 
2.17.1

