From 3d7dc267505e8db7a6fba9b08f73fa6c6b65eeff Mon Sep 17 00:00:00 2001
From: rajucm <raju.mallikarjun.chegaraddi@intel.com>
Date: Tue, 22 Aug 2023 04:28:42 +0000
Subject: [PATCH] mixinspec: enable swap-partition

Signed-off-by: rajucm <raju.mallikarjun.chegaraddi@intel.com>
---
 caas/mixins.spec | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/caas/mixins.spec b/caas/mixins.spec
index 564a542..db9923a 100755
--- a/caas/mixins.spec
+++ b/caas/mixins.spec
@@ -70,7 +70,8 @@ public-libraries: true
 device-specific: caas
 hdcpd: true
 treble: true
-swap: zram_auto(size=1073741824,swappiness=true,hardware=gordon_peak,disk_based_swap=true)
+#swap: zram(size=1073741824,swappiness=true,hardware=caas)
+swap-partition: true
 art-config: true
 debugfs: true
 disk-encryption: true
-- 
2.39.2

