From 6e3d5b5071c1850f6d9c2e49a4e92c712f8a002b Mon Sep 17 00:00:00 2001
From: Junjie Mao <junjie.mao@intel.com>
Date: Wed, 30 Aug 2023 01:54:56 -0400
Subject: [PATCH 11/14] virtio_ivshmem: exchange ivpos in virtio_header

Signed-off-by: Junjie Mao <junjie.mao@intel.com>
---
 drivers/virtio/virtio_ivshmem.c | 35 +++++++++++++++++++++++++--------
 1 file changed, 27 insertions(+), 8 deletions(-)

diff --git a/drivers/virtio/virtio_ivshmem.c b/drivers/virtio/virtio_ivshmem.c
index db6ac2c44dc0..fe3548b38363 100644
--- a/drivers/virtio/virtio_ivshmem.c
+++ b/drivers/virtio/virtio_ivshmem.c
@@ -24,6 +24,8 @@
 
 #define VIRTIO_STATE_READY	cpu_to_le32(1)
 
+#define FRONTEND_FLAG_PRESENT   cpu_to_le16(1)
+
 struct virtio_ivshmem_header {
 	__le32 revision;
 	__le32 size;
@@ -39,7 +41,21 @@ struct virtio_ivshmem_header {
 	};
 	__u8 config_event;
 	__u8 queue_event;
-	__u8 __rsvd[10];
+	__u8 __rsvd[2];
+	union {
+		__le32 frontend_status;
+		struct {
+			__le16 frontend_flags;
+			__le16 frontend_id;
+		};
+	};
+	union {
+		__le32 backend_status;
+		struct {
+			__le16 backend_flags;
+			__le16 backend_id;
+		};
+	};
 
 	struct virtio_pci_common_cfg common_config;
 	__u8 config[];
@@ -894,12 +910,6 @@ static int virtio_ivshmem_probe(struct pci_dev *pci_dev,
 	vi_dev->ivshm_regs = pcim_iomap_table(pci_dev)[0];
 
 	id = readl(&vi_dev->ivshm_regs->ivpos);
-	if (id > 1) {
-		dev_err(&pci_dev->dev, "invalid ID %d\n", id);
-		return -EINVAL;
-	}
-
-	vi_dev->peer_id = !id;
 
 	section_sz = pci_resource_len(pci_dev, 2);
 	if (section_sz == 0) {
@@ -937,6 +947,13 @@ static int virtio_ivshmem_probe(struct pci_dev *pci_dev,
 		dev_err(&pci_dev->dev, "invalid virtio-ivshmem revision\n");
 		return -EINVAL;
 	}
+	if (vi_dev->virtio_header->backend_flags == 0) {
+		dev_err(&pci_dev->dev, "backend is not present\n");
+		return -EINVAL;
+	}
+	vi_dev->peer_id = vi_dev->virtio_header->backend_id;
+	vi_dev->virtio_header->frontend_status = (id << 16) | FRONTEND_FLAG_PRESENT;
+
 	vi_dev->vdev.id.device = vi_dev->virtio_header->device_id;
 	vi_dev->vdev.id.vendor = vi_dev->virtio_header->vendor_id;
 
@@ -989,7 +1006,9 @@ static void virtio_ivshmem_remove(struct pci_dev *pci_dev)
 }
 
 static const struct pci_device_id virtio_ivshmem_id_table[] = {
-	{ PCI_DEVICE(PCI_VENDOR_ID_REDHAT_QUMRANET, 0x1110) },
+	//{ PCI_DEVICE(PCI_VENDOR_ID_REDHAT_QUMRANET, 0x1110) },
+	{ PCI_DEVICE(PCI_VENDOR_ID_REDHAT_QUMRANET, 0x1110),
+	  (PCI_CLASS_OTHERS << 16) | IVSHM_PROTO_VIRTIO_FRONT, 0xffff00 },
 	{ 0 }
 };
 
-- 
2.17.1

