From 9331474178b52a92769889bb43838ef81a402c5e Mon Sep 17 00:00:00 2001
From: ahs <amrita.h.s@intel.com>
Date: Tue, 17 Mar 2020 17:40:48 +0530
Subject: REVERTME: Disable nn-hal to build Celadon R

Tracked-On:
Signed-off-by: ahs <amrita.h.s@intel.com>
---
 Android.mk | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Android.mk b/Android.mk
index b3702e5..36bbde1 100644
--- a/Android.mk
+++ b/Android.mk
@@ -1,4 +1,4 @@
-LOCAL_PATH := $(call my-dir)
+#LOCAL_PATH := $(call my-dir)
 
-include $(LOCAL_PATH)/intel_nn_hal/Android.mk
+#include $(LOCAL_PATH)/intel_nn_hal/Android.mk
 
-- 
2.24.0

