From 06e8c6928b2a34a46c7e95c4d0a12c51a9a9cdc9 Mon Sep 17 00:00:00 2001
From: "Wan, Hao" <haox.wan@intel.com>
Date: Fri, 10 Nov 2023 15:42:13 +0800
Subject: [PATCH] Disable omx codec in goldfish

Signed-off-by: Chen, Tianmi <tianmi.chen@intel.com>
Signed-off-by: Wan, Hao <haox.wan@intel.com>
---
 Android.mk | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/Android.mk b/Android.mk
index 85b90d70..d6b1b567 100644
--- a/Android.mk
+++ b/Android.mk
@@ -172,10 +172,10 @@ ifeq ($(shell test $(PLATFORM_SDK_VERSION) -gt 28 -o $(IS_AT_LEAST_QPR1) = true
     include $(GOLDFISH_OPENGL_PATH)/system/hwc2/Android.mk
     include $(GOLDFISH_OPENGL_PATH)/system/hwc3/Android.mk
     # hardware codecs enabled after P
-    include $(GOLDFISH_OPENGL_PATH)/system/codecs/omx/common/Android.mk
-    include $(GOLDFISH_OPENGL_PATH)/system/codecs/omx/plugin/Android.mk
-    include $(GOLDFISH_OPENGL_PATH)/system/codecs/omx/avcdec/Android.mk
-    include $(GOLDFISH_OPENGL_PATH)/system/codecs/omx/vpxdec/Android.mk
+    # include $(GOLDFISH_OPENGL_PATH)/system/codecs/omx/common/Android.mk
+    # include $(GOLDFISH_OPENGL_PATH)/system/codecs/omx/plugin/Android.mk
+    # include $(GOLDFISH_OPENGL_PATH)/system/codecs/omx/avcdec/Android.mk
+    # include $(GOLDFISH_OPENGL_PATH)/system/codecs/omx/vpxdec/Android.mk
 endif
 
 endif
-- 
2.34.1

