From 86b383f2797cfc385f78e5da8cdee7fee7915cf2 Mon Sep 17 00:00:00 2001
From: shivasku82 <shiva.kumara.rudrappa@intel.com>
Date: Thu, 24 Aug 2023 13:48:22 +0530
Subject: [PATCH] add sepolicy for ivi camera service

---
 camera-ext/ivi/file_contexts | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/camera-ext/ivi/file_contexts b/camera-ext/ivi/file_contexts
index 4c93e38..bee4438 100644
--- a/camera-ext/ivi/file_contexts
+++ b/camera-ext/ivi/file_contexts
@@ -1,2 +1,4 @@
 /dev/media[0-9]+          u:object_r:video_device:s0
 /(vendor|system/vendor)/bin/android\.hardware\.automotive\.evs@1\.[0-9]-sample  u:object_r:hal_evs_default_exec:s0
+/(vendor|system/vendor)/bin/hw/android\.hardware\.camera\.provider@2\.[0-9]+-ivi-service       u:object_r:hal_camera_default_exec:s0
+
-- 
2.17.1

