From a348305ebe44cc43fd0d731d3c2787cacce278be Mon Sep 17 00:00:00 2001
From: jaishank <jaishankar.rajendran@intel.com>
Date: Fri, 18 Oct 2019 13:29:10 +0530
Subject: [PATCH] hwui

Tracked-On:
Signed-off-by: jaishank <jaishankar.rajendran@intel.com>
---
 libs/hwui/Android.bp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libs/hwui/Android.bp b/libs/hwui/Android.bp
index ebba4cb..9246f24 100644
--- a/libs/hwui/Android.bp
+++ b/libs/hwui/Android.bp
@@ -28,6 +28,8 @@ cc_defaults {
 
         // clang's warning is broken, see: https://llvm.org/bugs/show_bug.cgi?id=21629
         "-Wno-missing-braces",
+        "-mavx2",
+        "-mfma",
     ],
 
     include_dirs: [
-- 
2.7.4

