From 858433b77fce937bfeb318a64e682dc5dc905c52 Mon Sep 17 00:00:00 2001
From: Ming Tan <ming.tan@intel.com>
Date: Wed, 26 Dec 2018 13:05:00 +0800
Subject: [PATCH] Do not set ro.zygote to zygote32.

Signed-off-by: Ming Tan <ming.tan@intel.com>
---
 target/product/core_minimal.mk | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/target/product/core_minimal.mk b/target/product/core_minimal.mk
index 85646c1..c2fc72e 100644
--- a/target/product/core_minimal.mk
+++ b/target/product/core_minimal.mk
@@ -150,9 +150,9 @@ PRODUCT_PACKAGES += \
     fsck.f2fs \
     make_f2fs \
 
-PRODUCT_DEFAULT_PROPERTY_OVERRIDES += \
+#PRODUCT_DEFAULT_PROPERTY_OVERRIDES += \
     ro.zygote=zygote32
-PRODUCT_COPY_FILES += \
+#PRODUCT_COPY_FILES += \
     system/core/rootdir/init.zygote32.rc:root/init.zygote32.rc
 
 PRODUCT_COPY_FILES += \
-- 
1.9.1

