From f724bab492582d190393c69111b82297f4b8b1e5 Mon Sep 17 00:00:00 2001
From: Marc Mao <marc.mao@intel.com>
Date: Mon, 21 Nov 2022 10:46:25 +0800
Subject: [FEATURE] update mesa version string

Change-Id: I1f5ba5464b99d194e87a7672c6371846a6a4d8bd
Signed-off-by: Marc Mao <marc.mao@intel.com>
---
 VERSION | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/VERSION b/VERSION
index f3292f2dc8c..1c23a077683 100644
--- a/VERSION
+++ b/VERSION
@@ -1 +1 @@
-23.2.0-devel
+23.2.0-devel dg2-20230414
-- 
2.25.1

