From 5aac1ae8c114fe61a74157ae080a929248db87c5 Mon Sep 17 00:00:00 2001
From: renchenglei <chenglei.ren@intel.com>
Date: Tue, 9 Mar 2021 21:12:56 +0800
Subject: [PATCH 01/14] UPSTREAM: [common] fix build for newer versions of
 binutils

---
 common/src/crash.linux.c | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/common/src/crash.linux.c b/common/src/crash.linux.c
index 218d65f..72af903 100644
--- a/common/src/crash.linux.c
+++ b/common/src/crash.linux.c
@@ -72,7 +72,6 @@ static void load_symbols()
     DEBUG_ERROR("executable is not a bfd_object");
     return;
   }
-
   crash.section = bfd_get_section_by_name(crash.fd, ".text");
   if (!crash.section)
   {
@@ -98,10 +97,18 @@ static void load_symbols()
 
 static bool lookup_address(bfd_vma pc, const char ** filename, const char ** function, unsigned int * line, unsigned int * discriminator)
 {
+#ifdef bfd_get_section_flags
   if ((bfd_get_section_flags(crash.fd, crash.section) & SEC_ALLOC) == 0)
+#else
+  if ((bfd_section_flags(crash.section) & SEC_ALLOC) == 0)
+#endif
     return false;
 
+#ifdef bfd_get_section_size
   bfd_size_type size = bfd_get_section_size(crash.section);
+#else
+  bfd_size_type size = bfd_section_size(crash.section);
+#endif
   if (pc >= size)
     return false;
 
@@ -230,4 +237,4 @@ bool installCrashHandler(const char * exe)
   return true;
 }
 
-#endif
\ No newline at end of file
+#endif
-- 
2.25.1

