From 6417dce8f9b80da2f86d5163ca7e1eb1322204a5 Mon Sep 17 00:00:00 2001
From: renchenglei <chenglei.ren@intel.com>
Date: Wed, 7 Apr 2021 05:58:02 +0800
Subject: [PATCH 04/14] INTERNAL: Add support for dynamic window name

---
 client/src/config.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/client/src/config.c b/client/src/config.c
index 9c683c9..5735319 100644
--- a/client/src/config.c
+++ b/client/src/config.c
@@ -106,6 +106,7 @@ static struct Option options[] =
     .module         = "win",
     .name           = "title",
     .description    = "The window title",
+    .shortopt       = 't',
     .type           = OPTION_TYPE_STRING,
     .value.x_string = "Looking Glass (client)"
   },
-- 
2.25.1

