From 537c705cf4ee310c30f155b72b41e296d4db1072 Mon Sep 17 00:00:00 2001
From: renchenglei <chenglei.ren@intel.com>
Date: Thu, 22 Apr 2021 06:17:45 +0800
Subject: [PATCH 05/14] INTERNAL: Flag LG client is ready to receive frame

---
 client/src/main.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/client/src/main.c b/client/src/main.c
index f99277c..7b86914 100644
--- a/client/src/main.c
+++ b/client/src/main.c
@@ -284,6 +284,9 @@ static int frameThread(void * unused)
 
   while(state.running)
   {
+    //Flag LG client is ready
+    if(!(state.shm->flags & KVMFR_HEADER_FLAG_READY))
+      __sync_or_and_fetch(&state.shm->flags, KVMFR_HEADER_FLAG_READY);
     // poll until we have a new frame
     while(!(state.shm->frame.flags & KVMFR_FRAME_FLAG_UPDATE))
     {
-- 
2.25.1

