From 05c8e9035e3d77050147d569fdb2a445b4d403e5 Mon Sep 17 00:00:00 2001
From: Xihua Chen <xihua.chen@intel.com>
Date: Tue, 25 Sep 2018 14:53:00 +0800
Subject: [PATCH] add kernel 4.19 support in compatiblity check

Change-Id: I1fed1e0351ced4f256f2b58a376cff9e3f45b9ea
Tracked-On:
Signed-off-by: Xihua Chen <xihua.chen@intel.com>
---

diff --git a/compatibility_matrices/Android.mk b/compatibility_matrices/Android.mk
index 3702cf0..3e033a5 100644
--- a/compatibility_matrices/Android.mk
+++ b/compatibility_matrices/Android.mk
@@ -67,6 +67,7 @@
     4.4.107:$(my_kernel_config_data)/p/android-4.4 \
     4.9.84:$(my_kernel_config_data)/p/android-4.9 \
     4.14.42:$(my_kernel_config_data)/p/android-4.14 \
+    4.19.0:$(my_kernel_config_data)/p/android-4.19 \
 
 include $(BUILD_FRAMEWORK_COMPATIBILITY_MATRIX)
 
