From 6173d112113aa98cf3a41aa74358ea810ba96cff Mon Sep 17 00:00:00 2001
From: Kalyan Kondapally <kalyan.kondapally@intel.com>
Date: Tue, 25 Oct 2016 12:47:29 -0700
Subject: [PATCH 2/2] Build fixes.

---
 hdmi/tinyaudio_hw.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/hdmi/tinyaudio_hw.c b/hdmi/tinyaudio_hw.c
index 4aaf6f7..dfd8dfc 100644
--- a/hdmi/tinyaudio_hw.c
+++ b/hdmi/tinyaudio_hw.c
@@ -247,7 +247,7 @@ static int start_output_stream(struct stream_out *out)
     out->pcm_config.silence_threshold = 0;

     if(activePcm){
-      ALOGV("Closing already open tiny alsa stream running state %d",(int)(activePcm));
+      ALOGV("Closing already open tiny alsa stream running state %p", activePcm);
       pcm_close(activePcm);
       activePcm = NULL;
     }
@@ -269,7 +269,7 @@ static int start_output_stream(struct stream_out *out)
     activePcm = out->pcm;
     activeChannel = out->pcm_config.channels;

-    ALOGV("Initialized PCM device for channels %d handle = %d",out->pcm_config.channels, (int)activePcm);
+    ALOGI("Initialized PCM device for channels %d handle = %p", out->pcm_config.channels, activePcm);
     ALOGV("%s exit",__func__);
     return 0;
 }
@@ -310,7 +310,7 @@ static size_t out_get_buffer_size(const struct audio_stream *stream)
        buf_size/=LATENCY_TO_BUFFER_SIZE_RATIO;
     }

-    ALOGV("%s : %d, period_size : %d, frame_size : %d",
+    ALOGV("%s : %zu, period_size : %u, frame_size : %zu",
         __func__,
         buf_size,
         out->pcm_config.period_size,
@@ -569,7 +569,7 @@ static ssize_t out_write(struct audio_stream_out *stream, const void* buffer,
     int32_t* dstbuff = NULL;
     int outbytes = 0;

-    ALOGV("%s enter for bytes = %d channels = %d",__func__,bytes, out->pcm_config.channels);
+    ALOGV("%s enter for bytes = %zu channels = %u", __func__, bytes, out->pcm_config.channels);

     pthread_mutex_lock(&out->dev->lock);
     pthread_mutex_lock(&out->lock);
@@ -615,7 +615,7 @@ static ssize_t out_write(struct audio_stream_out *stream, const void* buffer,
     else
       ret = pcm_write(out->pcm, (void *)buffer, bytes);

-    ALOGV("pcm_write: %s done for %d input bytes, output bytes = %d ", pcm_get_error(out->pcm),bytes,outbytes);
+    ALOGV("pcm_write: %s done for %zu input bytes, output bytes = %d ", pcm_get_error(out->pcm), bytes, outbytes);

     free(dstbuff);

@@ -752,7 +752,7 @@ static int adev_open_output_stream(struct audio_hw_device *dev,
     pthread_mutex_lock(&out->lock);

     if(activePcm){
-      ALOGV("Closing already open tiny alsa stream %d",(int)out->pcm);
+      ALOGV("Closing already open tiny alsa stream %p", out->pcm);
       pcm_close(activePcm);
       activePcm = NULL;
     }
--
2.7.4
