From 3fa1960733d4533c3997091feebfe954be1e5efd Mon Sep 17 00:00:00 2001
From: rnaidu <ramya.v.naidu@intel.com>
Date: Fri, 10 Jul 2020 10:52:21 +0530
Subject: [PATCH] Add PRODUCT_STATIC_BOOT_CONTROL_HAL := bootctrl.intel.static

Tracked-On: OAM-91670
Signed-off-by: sgnanase <sundar.gnanasekaran@intel.com>
---
 caas/BoardConfig.mk | 1 +
 1 file changed, 1 insertion(+)

diff --git a/caas/BoardConfig.mk b/caas/BoardConfig.mk
index 685339a..097daa3 100644
--- a/caas/BoardConfig.mk
+++ b/caas/BoardConfig.mk
@@ -24,6 +24,7 @@ AB_OTA_PARTITIONS := \
 TARGET_NO_RECOVERY := true
 BOARD_USES_RECOVERY_AS_BOOT := true
 BOARD_SLOT_AB_ENABLE := true
+PRODUCT_STATIC_BOOT_CONTROL_HAL := bootctrl.intel.static
 BOARD_KERNEL_CMDLINE += rootfstype=ext4
 
 AB_OTA_POSTINSTALL_CONFIG += \
-- 
2.17.1

