From 1bedbdb197005f460b1926e8470e92cde30ecd54 Mon Sep 17 00:00:00 2001
From: renchenglei <chenglei.ren@intel.com>
Date: Sun, 14 Mar 2021 20:18:35 +0800
Subject: [PATCH] Enable uio

Tracked-On:
---
 caas/mixins.spec | 1 +
 1 file changed, 1 insertion(+)

diff --git a/caas/mixins.spec b/caas/mixins.spec
index a687f9867a67..690598a13797 100755
--- a/caas/mixins.spec
+++ b/caas/mixins.spec
@@ -89,3 +89,4 @@ suspend: auto
 sensors: mediation
 bugreport: true
 mainline-mod: true
+uio: true
-- 
2.17.1

