From 9878b2679991064ba4930adf50c7e2e94404003e Mon Sep 17 00:00:00 2001
From: tingai <ting.a.ai@intel.com>
Date: Tue, 20 Sep 2022 03:45:38 -0400
Subject: [PATCH] Make bootchart work on android S

Signed-off-by: tingai <ting.a.ai@intel.com>
---
 rootdir/init.rc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rootdir/init.rc b/rootdir/init.rc
index cd71aa8aa..a2a1a2878 100644
--- a/rootdir/init.rc
+++ b/rootdir/init.rc
@@ -678,7 +678,6 @@ on post-fs-data
     # Start bootcharting as soon as possible after the data partition is
     # mounted to collect more data.
     mkdir /data/bootchart 0755 shell shell encryption=Require
-    bootchart start
 
     # Avoid predictable entropy pool. Carry over entropy from previous boot.
     copy /data/system/entropy.dat /dev/urandom
@@ -1010,6 +1009,7 @@ on post-fs-data
     # module, delete if not.
     exec - system system -- /system/bin/tzdatacheck /apex/com.android.tzdata/etc/tz /data/misc/zoneinfo
 
+    bootchart start
     # sys.memfd_use set to false by default, which keeps it disabled
     # until it is confirmed that apps and vendor processes don't make
     # IOCTLs on ashmem fds any more.
-- 
2.25.1

