From d30cbad1fcb343fbb7aead53ba6c4f820cfac37d Mon Sep 17 00:00:00 2001
From: tingai <ting.a.ai@intel.com>
Date: Wed, 21 Sep 2022 01:35:14 -0400
Subject: [PATCH] Enable bootchart

Signed-off-by: tingai <ting.a.ai@intel.com>
---
 rootdir/init.rc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rootdir/init.rc b/rootdir/init.rc
index d10689a01..39b87b263 100644
--- a/rootdir/init.rc
+++ b/rootdir/init.rc
@@ -619,7 +619,6 @@ on post-fs-data
     # Start bootcharting as soon as possible after the data partition is
     # mounted to collect more data.
     mkdir /data/bootchart 0755 shell shell encryption=Require
-    bootchart start
 
     # Avoid predictable entropy pool. Carry over entropy from previous boot.
     copy /data/system/entropy.dat /dev/urandom
@@ -926,6 +925,7 @@ on post-fs-data
     # module, delete if not.
     exec - system system -- /system/bin/tzdatacheck /apex/com.android.tzdata/etc/tz /data/misc/zoneinfo
 
+    bootchart start
     # If there is no post-fs-data action in the init.<device>.rc file, you
     # must uncomment this line, otherwise encrypted filesystems
     # won't work.
-- 
2.25.1

