From ac62d0217f4927af6e994b3e596dc1b4e6866f9e Mon Sep 17 00:00:00 2001
From: "JianFeng,Zhou" <jianfeng.zhou@intel.com>
Date: Fri, 24 Apr 2020 16:14:12 +0800
Subject: [PATCH] ueventd: add rw attribe for block device /dev/block/vda

Tracked-On:
Signed-off-by: JianFeng,Zhou <jianfeng.zhou@intel.com>
---
 rootdir/ueventd.rc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/rootdir/ueventd.rc b/rootdir/ueventd.rc
index 09e5ed5..233e38a 100644
--- a/rootdir/ueventd.rc
+++ b/rootdir/ueventd.rc
@@ -41,6 +41,8 @@ subsystem sound
 
 /dev/pmsg0                0222   root       log
 
+/dev/block/vdb            0666   root root
+
 # kms driver for drm based gpu
 /dev/dri/*                0666   root       graphics
 
-- 
2.7.4

