From b559c4d465b42af841e8de4097ae54dc18ad9941 Mon Sep 17 00:00:00 2001
From: Shaofeng Tang <shaofeng.tang@intel.com>
Date: Thu, 2 Dec 2021 11:45:04 +0800
Subject: [PATCH] Debug log for 10bit video format

Signed-off-by: Shaofeng Tang <shaofeng.tang@intel.com>
---
 libs/renderengine/gl/GLESRenderEngine.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/libs/renderengine/gl/GLESRenderEngine.cpp b/libs/renderengine/gl/GLESRenderEngine.cpp
index 467f848237..f5b37a45c6 100644
--- a/libs/renderengine/gl/GLESRenderEngine.cpp
+++ b/libs/renderengine/gl/GLESRenderEngine.cpp
@@ -761,6 +761,7 @@ std::shared_ptr<ImageManager::Barrier> GLESRenderEngine::cacheExternalTextureBuf
 }
 
 status_t GLESRenderEngine::cacheExternalTextureBufferInternal(const sp<GraphicBuffer>& buffer) {
+    ALOGE("cacheExternalTextureBufferInternal begin");
     if (buffer == nullptr) {
         return BAD_VALUE;
     }
@@ -779,6 +780,10 @@ status_t GLESRenderEngine::cacheExternalTextureBufferInternal(const sp<GraphicBu
 
     bool created = newImage->setNativeWindowBuffer(buffer->getNativeBuffer(),
                                                    buffer->getUsage() & GRALLOC_USAGE_PROTECTED);
+    ALOGE("Invoking setNativeWindowBuffer() with size=%ux%u st=%u usage=%#" PRIx64 " fmt=%d",
+              buffer->getWidth(), buffer->getHeight(), buffer->getStride(), buffer->getUsage(),
+              buffer->getPixelFormat());
+
     if (!created) {
         ALOGE("Failed to create image. id=%" PRIx64 " size=%ux%u st=%u usage=%#" PRIx64 " fmt=%d",
               buffer->getId(), buffer->getWidth(), buffer->getHeight(), buffer->getStride(),
-- 
2.29.2

