From f970791acab6ffb1270f11ce0999f79babab85b4 Mon Sep 17 00:00:00 2001
From: Shaofeng Tang <shaofeng.tang@intel.com>
Date: Wed, 24 Nov 2021 14:49:08 +0800
Subject: [PATCH] Debug log for 8bit video format

Signed-off-by: Shaofeng Tang <shaofeng.tang@intel.com>
---
 libs/renderengine/gl/GLESRenderEngine.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/libs/renderengine/gl/GLESRenderEngine.cpp b/libs/renderengine/gl/GLESRenderEngine.cpp
index 0285c2f6f..edb7b5142 100644
--- a/libs/renderengine/gl/GLESRenderEngine.cpp
+++ b/libs/renderengine/gl/GLESRenderEngine.cpp
@@ -706,6 +706,7 @@ std::shared_ptr<ImageManager::Barrier> GLESRenderEngine::cacheExternalTextureBuf
 }
 
 status_t GLESRenderEngine::cacheExternalTextureBufferInternal(const sp<GraphicBuffer>& buffer) {
+    ALOGE("cacheExternalTextureBufferInternal begin");
     if (buffer == nullptr) {
         return BAD_VALUE;
     }
@@ -722,6 +723,10 @@ status_t GLESRenderEngine::cacheExternalTextureBufferInternal(const sp<GraphicBu
     // Create the image without holding a lock so that we don't block anything.
     std::unique_ptr<Image> newImage = createImage();
 
+    ALOGE("Invoking setNativeWindowBuffer() with size=%ux%u st=%u usage=%#" PRIx64 " fmt=%d",
+              buffer->getWidth(), buffer->getHeight(), buffer->getStride(), buffer->getUsage(),
+              buffer->getPixelFormat());
+
     bool created = newImage->setNativeWindowBuffer(buffer->getNativeBuffer(),
                                                    buffer->getUsage() & GRALLOC_USAGE_PROTECTED);
     if (!created) {
-- 
2.29.2

