From 14ae0dd6fb3838c34c068a1a50646c1f540e400f Mon Sep 17 00:00:00 2001
From: Fei Jiang <fei.jiang@intel.com>
Date: Fri, 22 Jul 2022 19:35:57 +0800
Subject: [PATCH] hack display type as internal

Signed-off-by: Fei Jiang <fei.jiang@intel.com>
---
 services/surfaceflinger/DisplayHardware/HWComposer.cpp | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/services/surfaceflinger/DisplayHardware/HWComposer.cpp b/services/surfaceflinger/DisplayHardware/HWComposer.cpp
index 7e45dabdea..415b54de97 100644
--- a/services/surfaceflinger/DisplayHardware/HWComposer.cpp
+++ b/services/surfaceflinger/DisplayHardware/HWComposer.cpp
@@ -373,9 +373,11 @@ std::optional<hal::HWConfigId> HWComposer::getActiveMode(PhysicalDisplayId displ
 
 ui::DisplayConnectionType HWComposer::getDisplayConnectionType(PhysicalDisplayId displayId) const {
     RETURN_IF_INVALID_DISPLAY(displayId, ui::DisplayConnectionType::Internal);
-    const auto& hwcDisplay = mDisplayData.at(displayId).hwcDisplay;
+    //const auto& hwcDisplay = mDisplayData.at(displayId).hwcDisplay;
 
-    ui::DisplayConnectionType type;
+    ui::DisplayConnectionType type = ui::DisplayConnectionType::Internal;
+    ALOGD("fei hack type as internal");
+/*
     const auto error = hwcDisplay->getConnectionType(&type);
 
     const auto FALLBACK_TYPE = hwcDisplay->getId() == mInternalHwcDisplayId
@@ -383,6 +385,7 @@ ui::DisplayConnectionType HWComposer::getDisplayConnectionType(PhysicalDisplayId
             : ui::DisplayConnectionType::External;
 
     RETURN_IF_HWC_ERROR(error, displayId, FALLBACK_TYPE);
+*/
     return type;
 }
 
-- 
2.32.0

