From 371118fbb353e352ead8419c8a99ac75bcd9f708 Mon Sep 17 00:00:00 2001
From: shivasku82 <shiva.kumara.rudrappa@intel.com>
Date: Mon, 2 Sep 2024 17:06:02 +0530
Subject: [PATCH] adding missing sepolicy for evs app

---
 car/evs_intel_app.te | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/car/evs_intel_app.te b/car/evs_intel_app.te
index a2e6a98..7ac5f08 100644
--- a/car/evs_intel_app.te
+++ b/car/evs_intel_app.te
@@ -12,7 +12,7 @@ init_daemon_domain(evs_intel_app)
 # gets access to its own files on disk
 type evs_intel_app_files, file_type, system_file_type;
 allow evs_intel_app evs_intel_app_files:file { getattr open read };
-allow evs_intel_app evs_intel_app_files:dir search;
+allow evs_intel_app evs_intel_app_files:dir { getattr search read };
 
 # Allow use of gralloc buffers and EGL
 allow evs_intel_app gpu_device:chr_file rw_file_perms;
-- 
2.17.1

