From 3283deef32d389ca59a27a57f68d6a06ef80b22b Mon Sep 17 00:00:00 2001
From: "Reddy, Alavala Srinivasa" <alavala.srinivasa.reddy@intel.com>
Date: Thu, 23 Nov 2023 21:12:44 +0530
Subject: [PATCH] Modify IA perf variant of library to have same name as
 original

Signed-off-by: bodapati <shalini.salomi.bodapati@intel.com>
---
 runtime/Android.bp | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/runtime/Android.bp b/runtime/Android.bp
index 5bb127fb1..0f27228b1 100644
--- a/runtime/Android.bp
+++ b/runtime/Android.bp
@@ -230,6 +230,9 @@ cc_library_shared {
 
 cc_library_shared {
     name: "libneuralnetworks_avx2",
+    host_supported: false,
+    override_lib_name: "libneuralnetworks",
+    relative_install_path: "IA-Perf/avx2",
     defaults: ["libneuralnetworks_generic"],
     min_sdk_version: "30",
     arch: {
@@ -240,11 +243,6 @@ cc_library_shared {
            cflags: [ "-mavx2", "-mfma", ],
         },
     },
-    target: {
-       android: {
-          relative_install_path: "IA-Perf/avx2",
-       },
-    },
 }
 
 // Required for tests (b/147158681)
-- 
2.17.1

