From 56858ee296e65a57c28500a822c1f6b5cacd8369 Mon Sep 17 00:00:00 2001
From: Vinay Prasad Kompella <vinay.kompella@intel.com>
Date: Tue, 17 May 2022 10:44:31 +0000
Subject: [PATCH] Expose sources for usage in libc

Signed-off-by: Vinay Prasad Kompella <vinay.kompella@intel.com>
---
 Android.bp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/Android.bp b/Android.bp
index 7f70060..8d9e99f 100644
--- a/Android.bp
+++ b/Android.bp
@@ -89,6 +89,13 @@ cc_library {
     ],
 }
 
+filegroup {
+    name: "libcpu_features_x86_sources",
+    srcs: [
+        "src/cpuinfo_x86.c",
+    ]
+}
+
 cc_library {
     name: "libcpu_features",
     defaults: [
-- 
2.33.1

