From cf2c44022b60074d5dd7b93898ceac2d391d0a7c Mon Sep 17 00:00:00 2001
From: anuvarsh <anuvarshini.bc@intel.com>
Date: Wed, 9 Oct 2019 13:46:44 +0530
Subject: [PATCH] App analysis

Tracked-On:
Signed-off-by: anuvarsh <anuvarshini.bc@intel.com>
---
 Android.bp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Android.bp b/Android.bp
index 85eaeb6..61b154c 100644
--- a/Android.bp
+++ b/Android.bp
@@ -15,6 +15,7 @@ cc_library {
         "-O3",
         "-DUSE_MMAP",
         "-DZLIB_CONST",
+	"-DFASTEST",
         "-Wall",
         "-Werror",
     ],
-- 
2.7.4

