From 8362fadfdd373c47240a1a12066f4c2623f5b53e Mon Sep 17 00:00:00 2001
From: Raveendra Babu Chennakesavulu <raveendra.babu.chennakesavulu@intel.com>
Date: Wed, 3 Jun 2020 15:12:16 +0530
Subject: [PATCH] [CIV] Making interface up and down to reflect proxy changes.

Tracked-On:

diff --git a/java/com/android/server/ethernet/EthernetNetworkFactory.java b/java/com/android/server/ethernet/EthernetNetworkFactory.java
index b8ed845..dd59404 100644
--- a/java/com/android/server/ethernet/EthernetNetworkFactory.java
+++ b/java/com/android/server/ethernet/EthernetNetworkFactory.java
@@ -46,6 +46,7 @@ import android.os.RemoteException;
 import android.text.TextUtils;
 import android.util.AndroidRuntimeException;
 import android.util.Log;
+import android.util.Slog;
 import android.util.SparseArray;
 
 import com.android.internal.util.IndentingPrintWriter;
@@ -377,6 +378,11 @@ public class EthernetNetworkFactory extends NetworkFactory {
 
         void setIpConfig(IpConfiguration ipConfig) {
             this.mIpConfig = ipConfig;
+            Slog.d(TAG, "Raveendra: setIpConfig: " +
+                          ipConfig);
+
+	    updateLinkState(false);
+	    updateLinkState(true);
         }
 
         boolean statisified(NetworkCapabilities requestedCapabilities) {
-- 
2.17.1

