From cd25cf8d82f4ddc18e97a7a4bdda22edb667b42c Mon Sep 17 00:00:00 2001
From: Raveendra Babu Chennakesavulu <raveendra.babu.chennakesavulu@intel.com>
Date: Fri, 24 Apr 2020 14:09:26 +0530
Subject: [PATCH] [REVEERT ME] Moving interface state down & up to after proxy
 config

Tracked-On:
Signed-off-by: Raveendra Babu Chennakesavulu
                        <raveendra.babu.chennakesavulu@intel.com>

diff --git a/java/com/android/server/ethernet/EthernetNetworkFactory.java b/java/com/android/server/ethernet/EthernetNetworkFactory.java
index b8ed845..058bbe0 100644
--- a/java/com/android/server/ethernet/EthernetNetworkFactory.java
+++ b/java/com/android/server/ethernet/EthernetNetworkFactory.java
@@ -377,6 +377,8 @@ public class EthernetNetworkFactory extends NetworkFactory {
 
         void setIpConfig(IpConfiguration ipConfig) {
             this.mIpConfig = ipConfig;
+	    updateLinkState(false);
+	    updateLinkState(true);
         }
 
         boolean statisified(NetworkCapabilities requestedCapabilities) {
@@ -552,6 +554,12 @@ public class EthernetNetworkFactory extends NetworkFactory {
 
         private static void provisionIpClient(IIpClient ipClient, IpConfiguration config,
                 String tcpBufferSizes) {
+	    //config.setProxySettings(ProxySettings.STATIC);
+	    //ProxyInfo mHttpProxy = new ProxyInfo("proxy01.iind.intel.com", 911, null);
+	    //config.setHttpProxy(mHttpProxy);
+	    //Slog.d(TAG, "Raveendra: post config proxysetting: " +
+	    //		    config.getProxySettings() + "proxy: " +
+	    //		    config.getHttpProxy());
             if (config.getProxySettings() == ProxySettings.STATIC ||
                     config.getProxySettings() == ProxySettings.PAC) {
                 try {
-- 
2.17.1

