From acf3813facdcde9de7378ba42f9c29fe85ea1985 Mon Sep 17 00:00:00 2001
From: Chen Lin Z <lin.z.chen@intel.com>
Date: Thu, 19 Sep 2019 11:10:41 +0800
Subject: [PATCH] compiler-rt: avx support

Tracked-On:
Signed-off-by: Chen Lin Z <lin.z.chen@intel.com>
---
 Android.bp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Android.bp b/Android.bp
index b8be740..049f2f3 100644
--- a/Android.bp
+++ b/Android.bp
@@ -123,7 +123,7 @@ cc_library {
         "lib/builtins/divsi3.c",
         "lib/builtins/divtf3.c",
         "lib/builtins/divti3.c",
-        "lib/builtins/divxc3.c",
+        //"lib/builtins/divxc3.c",
         "lib/builtins/eprintf.c",
         "lib/builtins/extenddftf2.c",
         "lib/builtins/extendhfsf2.c",
@@ -192,7 +192,7 @@ cc_library {
         "lib/builtins/mulvdi3.c",
         "lib/builtins/mulvsi3.c",
         "lib/builtins/mulvti3.c",
-        "lib/builtins/mulxc3.c",
+        //"lib/builtins/mulxc3.c",
         "lib/builtins/negdf2.c",
         "lib/builtins/negdi2.c",
         "lib/builtins/negsf2.c",
-- 
1.9.1

