From f4481164c63fcd76a57abc02c5a07ee8943957ed Mon Sep 17 00:00:00 2001
From: Basil Chew <basil.chew@intel.com>
Date: Tue, 31 Aug 2021 17:33:28 +0800
Subject: [PATCH 3/7] Disable EDID auto generation in QEMU

---
 hw/vfio/display.c | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/hw/vfio/display.c b/hw/vfio/display.c
index 89bc90508fb8..a84126bd7d55 100644
--- a/hw/vfio/display.c
+++ b/hw/vfio/display.c
@@ -25,6 +25,10 @@
 # define DRM_PLANE_TYPE_CURSOR  2
 #endif
 
+#ifndef DISABLE_EDID_AUTO_GENERATION
+#define DISABLE_EDID_AUTO_GENERATION
+#endif
+
 #define pread_field(_fd, _reg, _ptr, _fld)                              \
     (sizeof(_ptr->_fld) !=                                              \
      pread(_fd, &(_ptr->_fld), sizeof(_ptr->_fld),                      \
@@ -112,6 +116,10 @@ static int vfio_display_edid_ui_info(void *opaque, uint32_t idx,
     VFIOPCIDevice *vdev = opaque;
     VFIODisplay *dpy = vdev->dpy;
 
+    #if defined(DISABLE_EDID_AUTO_GENERATION)
+    return 0;
+    #endif
+
     if (!dpy->edid_regs) {
         return 0;
     }
@@ -131,6 +139,10 @@ static void vfio_display_edid_init(VFIOPCIDevice *vdev)
     int fd = vdev->vbasedev.fd;
     int ret;
 
+    #if defined(DISABLE_EDID_AUTO_GENERATION)
+    return;
+    #endif
+
     ret = vfio_get_dev_region_info(&vdev->vbasedev,
                                    VFIO_REGION_TYPE_GFX,
                                    VFIO_REGION_SUBTYPE_GFX_EDID,
@@ -179,6 +191,10 @@ err:
 
 static void vfio_display_edid_exit(VFIODisplay *dpy)
 {
+    #if defined(DISABLE_EDID_AUTO_GENERATION)
+    return;
+    #endif
+
     if (!dpy->edid_regs) {
         return;
     }
-- 
2.17.1

