From 0f98569de8d51a550229264b30924ef17c0c3904 Mon Sep 17 00:00:00 2001
From: HeYue <yue.he@intel.com>
Date: Mon, 12 Jun 2023 10:49:58 +0800
Subject: [PATCH] Disable EDID auto generation in QEMU

---
 hw/vfio/display.c | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/hw/vfio/display.c b/hw/vfio/display.c
index 78f4d82c1..1812e8915 100644
--- a/hw/vfio/display.c
+++ b/hw/vfio/display.c
@@ -25,6 +25,10 @@
 # define DRM_PLANE_TYPE_CURSOR  2
 #endif
 
+#ifndef DISABLE_EDID_AUTO_GENERATION
+#define DISABLE_EDID_AUTO_GENERATION
+#endif
+
 #define pread_field(_fd, _reg, _ptr, _fld)                              \
     (sizeof(_ptr->_fld) !=                                              \
      pread(_fd, &(_ptr->_fld), sizeof(_ptr->_fld),                      \
@@ -112,6 +116,10 @@ static void vfio_display_edid_ui_info(void *opaque, uint32_t idx,
     VFIOPCIDevice *vdev = opaque;
     VFIODisplay *dpy = vdev->dpy;
 
+    #if defined(DISABLE_EDID_AUTO_GENERATION)
+    return;
+    #endif
+
     if (!dpy->edid_regs) {
         return;
     }
@@ -129,6 +137,10 @@ static void vfio_display_edid_init(VFIOPCIDevice *vdev)
     int fd = vdev->vbasedev.fd;
     int ret;
 
+    #if defined(DISABLE_EDID_AUTO_GENERATION)
+    return;
+    #endif
+
     ret = vfio_get_dev_region_info(&vdev->vbasedev,
                                    VFIO_REGION_TYPE_GFX,
                                    VFIO_REGION_SUBTYPE_GFX_EDID,
@@ -177,6 +189,11 @@ err:
 
 static void vfio_display_edid_exit(VFIODisplay *dpy)
 {
+
+    #if defined(DISABLE_EDID_AUTO_GENERATION)
+    return;
+    #endif
+
     if (!dpy->edid_regs) {
         return;
     }
-- 
2.25.1

