From 86a0465b5065f69dd965efd94437b9858ba435ce Mon Sep 17 00:00:00 2001
From: yhe39 <yue.he@intel.com>
Date: Mon, 29 Nov 2021 11:11:27 +0800
Subject: [PATCH] Set the fourcc if it is 0

Signed-off-by: yhe39 <yue.he@intel.com>
---
 ui/egl-helpers.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/ui/egl-helpers.c b/ui/egl-helpers.c
index 42f0b45d4..e7d84febc 100644
--- a/ui/egl-helpers.c
+++ b/ui/egl-helpers.c
@@ -284,7 +284,11 @@ void egl_dmabuf_import_texture(QemuDmaBuf *dmabuf)
     attrs[i++] = EGL_HEIGHT;
     attrs[i++] = dmabuf->height;
     attrs[i++] = EGL_LINUX_DRM_FOURCC_EXT;
-    attrs[i++] = dmabuf->fourcc;
+    if (dmabuf->fourcc == 0) {
+        attrs[i++] = 0x34324241;
+    }
+    else
+        attrs[i++] = dmabuf->fourcc;
 
     attrs[i++] = EGL_DMA_BUF_PLANE0_FD_EXT;
     attrs[i++] = dmabuf->fd;
-- 
2.25.1

