From 29fca0fbc433eab170cf8dda0e46a85fef501d9a Mon Sep 17 00:00:00 2001
From: "Martin, Chen" <haochuan.z.chen@intel.com>
Date: Fri, 14 Jul 2023 17:23:36 -0400
Subject: [PATCH] fix AudioMixer resampler mono playback

---
 media/libaudioprocessing/AudioMixerBase.cpp | 22 +++++++++++++++------
 1 file changed, 16 insertions(+), 6 deletions(-)

diff --git a/media/libaudioprocessing/AudioMixerBase.cpp b/media/libaudioprocessing/AudioMixerBase.cpp
index 8d24d76ebd..472a7b2d83 100644
--- a/media/libaudioprocessing/AudioMixerBase.cpp
+++ b/media/libaudioprocessing/AudioMixerBase.cpp
@@ -1640,13 +1640,23 @@ AudioMixerBase::hook_t AudioMixerBase::TrackBase::getTrackHook(int trackType, ui
     case TRACKTYPE_RESAMPLEMONO:
         switch (mixerInFormat) {
         case AUDIO_FORMAT_PCM_FLOAT:
-            return (AudioMixerBase::hook_t) &TrackBase::track__Resample<
-                    MIXTYPE_STEREOEXPAND, float /*TO*/, float /*TI*/,
-                    TYPE_AUX>;
+            if (channelCount == FCC_2) {
+                return (AudioMixerBase::hook_t) &TrackBase::track__Resample<
+                        MIXTYPE_STEREOEXPAND, float /*TO*/, float /*TI*/,
+                        TYPE_AUX>;
+            } else {
+                return (AudioMixerBase::hook_t) &TrackBase::track__Resample<
+                        MIXTYPE_MULTI, float /*TO*/, float /*TI*/, TYPE_AUX>;
+            }
         case AUDIO_FORMAT_PCM_16_BIT:
-            return (AudioMixerBase::hook_t) &TrackBase::track__Resample<
-                    MIXTYPE_STEREOEXPAND, int32_t /*TO*/, int16_t /*TI*/,
-                    TYPE_AUX>;
+            if (channelCount == FCC_2) {
+                return (AudioMixerBase::hook_t) &TrackBase::track__Resample<
+                        MIXTYPE_STEREOEXPAND, int32_t /*TO*/, int16_t /*TI*/,
+                        TYPE_AUX>;
+            } else {
+                return (AudioMixerBase::hook_t) &TrackBase::track__Resample<
+                        MIXTYPE_MULTI, int32_t /*TO*/, int16_t /*TI*/, TYPE_AUX>;
+            }
         default:
             LOG_ALWAYS_FATAL("bad mixerInFormat: %#x", mixerInFormat);
             break;
-- 
2.17.1

