diff --git a/contents/namespace/sphal.cc b/contents/namespace/sphal.cc
index e97197f..cf9269b 100644
--- a/contents/namespace/sphal.cc
+++ b/contents/namespace/sphal.cc
@@ -47,6 +47,11 @@ Namespace BuildSphalNamespace([[maybe_unused]] const Context& ctx) {
   ns.AddPermittedPath("/vendor/${LIB}", AsanPath::WITH_DATA_ASAN);
   ns.AddPermittedPath("/system/vendor/${LIB}", AsanPath::NONE);
 
+  ns.AddSearchPath("/system/${LIB}/arm", AsanPath::WITH_DATA_ASAN);
+  ns.AddPermittedPath("/system/${LIB}/arm", AsanPath::WITH_DATA_ASAN);
+  ns.AddSearchPath("/system/${LIB}/arm/nb", AsanPath::WITH_DATA_ASAN);
+  ns.AddSearchPath("/system/${LIB}/arm/nb", AsanPath::WITH_DATA_ASAN);
+
   if (ctx.IsApexBinaryConfig() && !ctx.IsVndkAvailable()) {
     // If device is legacy, let Sphal libraries access to system lib path for
     // VNDK-SP libraries
diff --git a/contents/namespace/systemdefault.cc b/contents/namespace/systemdefault.cc
index cd73836..37a830a 100644
--- a/contents/namespace/systemdefault.cc
+++ b/contents/namespace/systemdefault.cc
@@ -67,6 +67,8 @@ Namespace BuildSystemDefaultNamespace([[maybe_unused]] const Context& ctx) {
         "/system/${LIB}/drm",
         "/system/${LIB}/extractors",
         "/system/${LIB}/hw",
+	   "/system/${LIB}/arm",
+	   "/system/${LIB}/arm/nb",
         system_ext + "/${LIB}",
 
         // These are where odex files are located. libart has to be able to
