diff --git a/contents/namespace/sphal.cc b/contents/namespace/sphal.cc
index e97197f..6ea729d 100644
--- a/contents/namespace/sphal.cc
+++ b/contents/namespace/sphal.cc
@@ -43,10 +43,20 @@ Namespace BuildSphalNamespace([[maybe_unused]] const Context& ctx) {
   ns.AddSearchPath("/vendor/${LIB}/egl", AsanPath::WITH_DATA_ASAN);
   ns.AddSearchPath("/vendor/${LIB}/hw", AsanPath::NONE);
 
+  ns.AddSearchPath("/vendor/${LIB}/arm", AsanPath::WITH_DATA_ASAN);
+  ns.AddPermittedPath("/vendor/${LIB}/arm", AsanPath::WITH_DATA_ASAN);
+  ns.AddSearchPath("/vendor/${LIB}/arm/nb", AsanPath::WITH_DATA_ASAN);
+  ns.AddSearchPath("/vendor/${LIB}/arm/nb", AsanPath::WITH_DATA_ASAN);
+
   ns.AddPermittedPath("/odm/${LIB}", AsanPath::WITH_DATA_ASAN);
   ns.AddPermittedPath("/vendor/${LIB}", AsanPath::WITH_DATA_ASAN);
   ns.AddPermittedPath("/system/vendor/${LIB}", AsanPath::NONE);
 
+  ns.AddSearchPath("/system/${LIB}/arm", AsanPath::WITH_DATA_ASAN);
+  ns.AddPermittedPath("/system/${LIB}/arm", AsanPath::WITH_DATA_ASAN);
+  ns.AddSearchPath("/system/${LIB}/arm/nb", AsanPath::WITH_DATA_ASAN);
+  ns.AddSearchPath("/system/${LIB}/arm/nb", AsanPath::WITH_DATA_ASAN);
+
   if (ctx.IsApexBinaryConfig() && !ctx.IsVndkAvailable()) {
     // If device is legacy, let Sphal libraries access to system lib path for
     // VNDK-SP libraries
diff --git a/contents/namespace/systemdefault.cc b/contents/namespace/systemdefault.cc
index cd73836..49b0b64 100644
--- a/contents/namespace/systemdefault.cc
+++ b/contents/namespace/systemdefault.cc
@@ -67,6 +67,11 @@ Namespace BuildSystemDefaultNamespace([[maybe_unused]] const Context& ctx) {
         "/system/${LIB}/drm",
         "/system/${LIB}/extractors",
         "/system/${LIB}/hw",
+        "/system/${LIB}/arm",
+        "/system/${LIB}/arm/nb",
+        "/vendor/${LIB}",
+        "/vendor/${LIB}/arm",
+        "/vendor/${LIB}/arm/nb",
         system_ext + "/${LIB}",
 
         // These are where odex files are located. libart has to be able to
