From 8ef81f1fb8bc1b42144674e72c43e71ecdb886ba Mon Sep 17 00:00:00 2001
From: rnaidu <ramya.v.naidu@intel.com>
Date: Tue, 31 Mar 2020 07:54:17 +0530
Subject: [PATCH] fixing the allowzip64 on celadon images

Tracked-On: N/A
Signed-off-by: rnaidu <ramya.v.naidu@intel.com>
---
 Lib/zipfile.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Lib/zipfile.py b/Lib/zipfile.py
index d2d3b69328..1bdd96bc58 100644
--- a/Lib/zipfile.py
+++ b/Lib/zipfile.py
@@ -1888,7 +1888,7 @@ class ZipFile:
             requires_zip64 = "Central directory size"
         if requires_zip64:
             # Need to write the ZIP64 end-of-archive records
-            if not self._allowZip64:
+            if self._allowZip64:
                 raise LargeZipFile(requires_zip64 +
                                    " would require ZIP64 extensions")
             zip64endrec = struct.pack(
-- 
2.24.1

