From a96722bd6e291192fc6ad7c7778306aacd497bbf Mon Sep 17 00:00:00 2001
From: Kalyan Kondapally <kalyan.kondapally@intel.com>
Date: Thu, 29 Sep 2016 22:25:12 -0700
Subject: [PATCH 4/7] symlink system/lib to /lib

---
 rootdir/init.rc | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/rootdir/init.rc b/rootdir/init.rc
index 56379db..c54b9b6 100644
--- a/rootdir/init.rc
+++ b/rootdir/init.rc
@@ -26,6 +26,9 @@ on early-init
     # Set the security context of /postinstall if present.
     restorecon /postinstall

+    # for /lib/firmware
+    symlink system/lib /lib
+
     start ueventd

 on init
--
2.7.4
