From 976cace77b090ccc79ebe49341bf797ab47644e5 Mon Sep 17 00:00:00 2001
From: anuvarsh <anuvarshini.bc@intel.com>
Date: Tue, 20 Aug 2019 17:00:13 +0530
Subject: [PATCH] rs android.bp

Tracked-On:
Signed-off-by: anuvarsh <anuvarshini.bc@intel.com>
---
 Android.bp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Android.bp b/Android.bp
index 08777529..2e909887 100644
--- a/Android.bp
+++ b/Android.bp
@@ -12,6 +12,8 @@ cc_defaults {
         "-Wextra",
         "-Wno-unused-parameter",
         "-Wno-unused-variable",
+	"-mavx2",
+	"-mfma",
     ],
 }
 
-- 
2.21.0

