From 65e20dba7e56bcffc1e4eddb778a62e1eb8301b1 Mon Sep 17 00:00:00 2001
From: anuvarsh <anuvarshini.bc@intel.com>
Date: Mon, 26 Aug 2019 11:57:15 +0530
Subject: [PATCH] rs_Android.bp

Tracked-On:
Signed-off-by: anuvarsh <anuvarshini.bc@intel.com>
---
 Android.bp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Android.bp b/Android.bp
index 08777529..da196361 100644
--- a/Android.bp
+++ b/Android.bp
@@ -12,6 +12,9 @@ cc_defaults {
         "-Wextra",
         "-Wno-unused-parameter",
         "-Wno-unused-variable",
+	"-mavx",
+	"-mavx2",
+	"-mfma",
     ],
 }
 
-- 
2.21.0

