From 66da93af8ee843191af6c362464ecb89ca1a02a0 Mon Sep 17 00:00:00 2001
From: anuvarsh <anuvarshini.bc@intel.com>
Date: Mon, 9 Sep 2019 12:11:32 +0530
Subject: [PATCH] cpuref ofast

Tracked-On:
Signed-off-by: anuvarsh <anuvarshini.bc@intel.com>
---
 cpu_ref/Android.bp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/cpu_ref/Android.bp b/cpu_ref/Android.bp
index f54f14a4..69b7b8d6 100644
--- a/cpu_ref/Android.bp
+++ b/cpu_ref/Android.bp
@@ -121,6 +121,7 @@ cc_library_shared {
 	"-mavx",
 	"-mavx2",
 	"-mfma",
+	"-Ofast",
     ],
 
     product_variables: {
-- 
2.23.0

