From 78b7416cc17d506f49ecb6d747b876de6b5b42c6 Mon Sep 17 00:00:00 2001
From: "Wang, Tao" <tao.wang@intel.com>
Date: Wed, 29 Nov 2023 13:22:46 +0800
Subject: [PATCH] Add sepolicy support for houdini debug

---
 houdini/file_contexts | 1 +
 1 file changed, 1 insertion(+)

diff --git a/houdini/file_contexts b/houdini/file_contexts
index 05ebd30..fac1400 100644
--- a/houdini/file_contexts
+++ b/houdini/file_contexts
@@ -1,6 +1,7 @@
 /(vendor|system/vendor)/lib(64)?/libhoudini\.so		u:object_r:same_process_hal_file:s0
 /(vendor|system/vendor)/lib/arm(/.*)?			u:object_r:same_process_hal_file:s0
 /(vendor|system/vendor)/lib64/arm64(/.*)?		u:object_r:same_process_hal_file:s0
+/(vendor|system/vendor)/lib(64)?/ubt\.config             u:object_r:same_process_hal_file:s0
 /vendor/bin/arm(64)?(/.*)?				u:object_r:same_process_hal_file:s0
 /vendor/bin/houdini(64)?           			u:object_r:same_process_hal_file:s0
 
-- 
2.34.1

