From c899837fceacb4ffc5b3a5980c00c7acfad26e8d Mon Sep 17 00:00:00 2001
From: HeYue <yue.he@intel.com>
Date: Mon, 31 Jul 2023 13:26:41 +0800
Subject: [PATCH] Add sepolicy for minigbm AIDL libs

Tracked-On: OAM-111525
Signed-off-by: HeYue <yue.he@intel.com>
---
 vendor/file_contexts | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/vendor/file_contexts b/vendor/file_contexts
index 7b0700121e16..7a9ec47f021e 100644
--- a/vendor/file_contexts
+++ b/vendor/file_contexts
@@ -1,4 +1,7 @@
 /vendor/bin/logwrapper      u:object_r:logwrapper_exec:s0
+/vendor/bin/hw/android\.hardware\.graphics\.allocator-service\.minigbm   u:object_r:hal_graphics_allocator_default_exec:s0
 /vendor/bin/hw/android\.hardware\.graphics\.allocator@4\.0-service\.minigbm_intel   u:object_r:hal_graphics_allocator_default_exec:s0
 /vendor/lib(64)?/hw/android\.hardware\.graphics\.mapper@4\.0-impl\.minigbm_intel\.so u:object_r:same_process_hal_file:s0
+/vendor/lib(64)?/hw/mapper\.minigbm\.so u:object_r:same_process_hal_file:s0
 /vendor/lib(64)?/libminigbm_gralloc_intel.so u:object_r:same_process_hal_file:s0
+/vendor/lib(64)?/libminigbm_gralloc4_utils.so  u:object_r:same_process_hal_file:s0
-- 
2.17.1

