From 4f9581e8258e18201847fb5cb63fbf397a5cc5e9 Mon Sep 17 00:00:00 2001
From: Shwetha B <shwetha.b@intel.com>
Date: Thu, 2 Jan 2025 11:56:44 +0530
Subject: [PATCH] Disable rtc during suspend

Signed-off-by: Shwetha B <shwetha.b@intel.com>
---
 kernel/time/alarmtimer.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/kernel/time/alarmtimer.c b/kernel/time/alarmtimer.c
index 4657cb8e8b1f..a787844245e5 100644
--- a/kernel/time/alarmtimer.c
+++ b/kernel/time/alarmtimer.c
@@ -923,7 +923,9 @@ static const struct dev_pm_ops alarmtimer_pm_ops = {
 static struct platform_driver alarmtimer_driver = {
 	.driver = {
 		.name = "alarmtimer",
+#if 0
 		.pm = &alarmtimer_pm_ops,
+#endif
 	}
 };
 
-- 
2.46.0

