From 4b8abf24accebe890ae85f418563ed05b6c62adb Mon Sep 17 00:00:00 2001
From: dengliqiang <liqiangx.deng@intel.com>
Date: Tue, 31 Dec 2019 07:00:42 +0800
Subject: [PATCH] rename hwc to intel for caas

Tracked-On: OAM-88753
Signed-off-by: mayifang <yifangx.ma@intel.com>
---
 Android.bp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Android.bp b/Android.bp
index 6c492d0..350bdac 100644
--- a/Android.bp
+++ b/Android.bp
@@ -92,7 +92,7 @@ cc_library_shared {
 }
 
 cc_library_shared {
-    name: "hwcomposer.broxton",
+    name: "hwcomposer.intel",
     defaults: ["hwcomposer.drm_defaults.intel"],
     whole_static_libs: ["drm_hwcomposer_intel"],
     srcs: [
-- 
2.24.0

