From 1ac798d79d0ff09a367eaca92ce22440028fdb72 Mon Sep 17 00:00:00 2001
From: yifang ma <yifangx.ma@intel.com>
Date: Sat, 11 Jan 2020 00:31:03 +0800
Subject: [PATCH] rename hwc to intel for caas

Tracked-On: OAM-88753
Signed-off-by: yifang ma <yifangx.ma@intel.com>
---
 Android.bp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Android.bp b/Android.bp
index 0e35f96..6eae578 100644
--- a/Android.bp
+++ b/Android.bp
@@ -92,7 +92,7 @@ cc_library_shared {
 }
 
 cc_library_shared {
-    name: "hwcomposer.broxton",
+    name: "hwcomposer.intel",
     defaults: ["hwcomposer.drm_defaults.intel"],
     whole_static_libs: ["drm_hwcomposer_intel"],
     cppflags: ["-DDRV_I915"],
-- 
2.24.0

