From c8e5f51a0b92b68adfe1adcf9d24298fae4859bc Mon Sep 17 00:00:00 2001
From: revatipx <revatix.prasad@intel.com>
Date: Mon, 24 Jun 2019 16:59:48 +0530
Subject: [PATCH] System : adb root

Tracked-On:
Signed-off-by: Prabhat Chand Pandey <prabhat.chand.pandey@intel.com>
---
 adb/daemon/usb.cpp | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/adb/daemon/usb.cpp b/adb/daemon/usb.cpp
index 60e05a8..0471f6a 100644
--- a/adb/daemon/usb.cpp
+++ b/adb/daemon/usb.cpp
@@ -29,6 +29,7 @@
 #include <sys/mman.h>
 #include <sys/types.h>
 #include <unistd.h>
+#include <linux/usb/xhcidbc.h>
 
 #include <algorithm>
 #include <atomic>
@@ -627,6 +628,16 @@ static int usb_dbc_read(usb_handle* h, void* data, int len) {
     D("[ done fd=%d ]", h->bulk_out);
     return 0;
 }
+static void usb_dbc_kick(usb_handle* h) {
+    int err;
+    D("usb_dbc_kick called\n");
+    err = ioctl(h->bulk_in, FLUSH_DBC_EP);
+    if (err < 0) {
+        D("[ kick: source (fd=%d) clear halt failed (%d) ]", h->bulk_in, errno);
+    }
+
+    h->kicked = true;
+}
 
 static void usb_dbc_close(usb_handle* h) {
     h->kicked = false;
@@ -646,6 +657,7 @@ static void usb_dbc_init() {
 
     h->write = usb_dbc_write;
     h->read = usb_dbc_read;
+    h->kick = usb_dbc_kick;
     h->close = usb_dbc_close;
 
     D("[ usb_init - starting thread ]");
-- 
2.7.4

