From 99b288fa8176917fe52241c63837c987f2953f8c Mon Sep 17 00:00:00 2001
From: user189 <shiva.kumara.rudrappa@intel.com>
Date: Wed, 10 Feb 2021 08:47:54 +0530
Subject: [PATCH] camera service remove

---
 camera/provider/2.4/default/Android.bp | 42 --------------------------
 1 file changed, 42 deletions(-)

diff --git a/camera/provider/2.4/default/Android.bp b/camera/provider/2.4/default/Android.bp
index 627ddf40b..9077e3e5c 100644
--- a/camera/provider/2.4/default/Android.bp
+++ b/camera/provider/2.4/default/Android.bp
@@ -129,49 +129,7 @@ cc_library_shared {
     export_include_dirs: ["."],
 }
 
-cc_defaults {
-    name: "camera_service_defaults",
-    defaults: ["hidl_defaults"],
-    proprietary: true,
-    relative_install_path: "hw",
-    srcs: ["service.cpp"],
-    shared_libs: [
-        "android.hardware.camera.common@1.0",
-        "android.hardware.camera.device@1.0",
-        "android.hardware.camera.device@3.2",
-        "android.hardware.camera.device@3.3",
-        "android.hardware.camera.device@3.4",
-        "android.hardware.camera.device@3.5",
-        "android.hardware.camera.provider@2.4",
-        "android.hardware.graphics.mapper@2.0",
-        "android.hardware.graphics.mapper@3.0",
-        "android.hardware.graphics.mapper@4.0",
-        "android.hidl.allocator@1.0",
-        "android.hidl.memory@1.0",
-        "libbinder",
-        "libcamera_metadata",
-        "libhardware",
-        "libhidlbase",
-        "liblog",
-        "libutils",
-    ],
-    static_libs: [
-        "android.hardware.camera.common@1.0-helper",
-    ],
-    header_libs: [
-        "camera.device@3.4-external-impl_headers",
-        "camera.device@3.4-impl_headers",
-        "camera.device@3.5-external-impl_headers",
-        "camera.device@3.5-impl_headers",
-    ],
-}
 
-cc_binary {
-    name: "android.hardware.camera.provider@2.4-service",
-    defaults: ["camera_service_defaults"],
-    compile_multilib: "32",
-    init_rc: ["android.hardware.camera.provider@2.4-service.rc"],
-}
 
 cc_binary {
     name: "android.hardware.camera.provider@2.4-service_64",
-- 
2.17.1

