From afe1d36aeedc9d10c4d5b2906ae247871ce1103d Mon Sep 17 00:00:00 2001
From: "Liu, Kai1" <kai1.liu@intel.com>
Date: Tue, 29 Aug 2023 10:49:16 +0800
Subject: [PATCH] Debug patch: print all actions of init

Signed-off-by: Liu, Kai1 <kai1.liu@intel.com>
---
 init/action.cpp | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/init/action.cpp b/init/action.cpp
index 1e998aeeb..8e41e8638 100644
--- a/init/action.cpp
+++ b/init/action.cpp
@@ -167,7 +167,14 @@ void Action::ExecuteCommand(const Command& command) const {
         std::string trigger_name = BuildTriggersString();
         std::string cmd_str = command.BuildCommandString();
 
-        LOG(INFO) << "Command '" << cmd_str << "' action=" << trigger_name << " (" << filename_
+        LOG(INFO) << "liukai1: Command '" << cmd_str << "' action=" << trigger_name << " (" << filename_
+                  << ":" << command.line() << ") took " << duration.count() << "ms and "
+                  << (result.ok() ? "succeeded" : "failed: " + result.error().message());
+    } else {
+        std::string trigger_name = BuildTriggersString();
+        std::string cmd_str = command.BuildCommandString();
+
+        LOG(INFO) << "liukai2: Command '" << cmd_str << "' action=" << trigger_name << " (" << filename_
                   << ":" << command.line() << ") took " << duration.count() << "ms and "
                   << (result.ok() ? "succeeded" : "failed: " + result.error().message());
     }
-- 
2.34.1

