From f2610231660c194b148db2451821f1f054204d67 Mon Sep 17 00:00:00 2001
From: Yong Yao <yong.yao@intel.com>
Date: Sun, 25 Feb 2018 11:38:18 -0800
Subject: [PATCH 2/2] [DO NOT MERGE] Use the private libdrm name

Change-Id: I98dd30a47c960e6052520383f0c3801566229a8b
Signed-off-by: Yong Yao <yong.yao@intel.com>
---
 Android.mk | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Android.mk b/Android.mk
index 0ee30cc..e92ab88 100644
--- a/Android.mk
+++ b/Android.mk
@@ -14,7 +14,7 @@ SUBDIRS := cros_gralloc
 LOCAL_SHARED_LIBRARIES := \
 	libcutils \
 	liblog \
-	libdrm \
+	libdrm_pri \
 	libsync
 
 LOCAL_SRC_FILES := \
@@ -53,7 +53,7 @@ LOCAL_C_INCLUDES += frameworks/native/libs/nativebase/include \
 ifneq ($(filter $(intel_drivers), $(BOARD_GPU_DRIVERS)),)
 LOCAL_CPPFLAGS += -DDRV_I915
 LOCAL_CFLAGS += -DDRV_I915
-LOCAL_SHARED_LIBRARIES += libdrm_intel
+LOCAL_SHARED_LIBRARIES += libdrm_intel_pri
 endif
 
 ifeq ($(shell test $(PLATFORM_SDK_VERSION) -ge 27; echo $$?), 0)
-- 
2.21.0

