From 74d248badb8aa85a519edaed8394728d2d95912a Mon Sep 17 00:00:00 2001
From: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
Date: Wed, 15 Sep 2021 21:45:54 +0530
Subject: [PATCH] Remove LLVM_IAS check as LLVM_IAS is not yet supported

Tracked-On: OAM-99477
Signed-off-by: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
---
 arch/Kconfig | 1 -
 1 file changed, 1 deletion(-)

diff --git a/arch/Kconfig b/arch/Kconfig
index 26916e0fe179..7a65e3476402 100644
--- a/arch/Kconfig
+++ b/arch/Kconfig
@@ -644,7 +644,6 @@ config HAS_LTO_CLANG
 	# Clang >= 11: https://github.com/ClangBuiltLinux/linux/issues/510
 	depends on CC_IS_CLANG && CLANG_VERSION >= 110000 && LD_IS_LLD
 	depends on $(success,test $(LLVM) -eq 1)
-	depends on $(success,test $(LLVM_IAS) -eq 1)
 	depends on $(success,$(NM) --help | head -n 1 | grep -qi llvm)
 	depends on $(success,$(AR) --help | head -n 1 | grep -qi llvm)
 	depends on ARCH_SUPPORTS_LTO_CLANG
-- 
2.33.0

