From 0a86a59a4c8b9c23d98fffc4ebe2518c3076f9cc Mon Sep 17 00:00:00 2001
From: reddyr3x <rakeshx.reddy@intel.com>
Date: Tue, 14 Aug 2018 15:38:37 +0530
Subject: [PATCH] Making modifications to Android.mk for Android-P celadon

Tracked-On: OAM-67515
Signed-off-by: yasoda <lab_aqa@intel.com>
---
 Android.mk | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/Android.mk b/Android.mk
index e4f4e74..8284304 100644
--- a/Android.mk
+++ b/Android.mk
@@ -18,6 +18,15 @@ LOCAL_PATH := $(call my-dir)
 
 art_path := $(LOCAL_PATH)
 
+# Allow vendor to deploy another ART VM with proprietary optimizations.
+ifneq ($(VENDOR_ART_PATH),)
+   # Set this way to reduce conflicts in file.
+   art_path := $(VENDOR_ART_PATH)
+else
+   # If vendor is not deploying separate VM, just use ART.
+   VENDOR_ART_PATH := $(art_path)
+endif
+
 ########################################################################
 # clean-oat rules
 #
-- 
1.9.1

