From 085b64bd40f3fef1725c449ada5402cc45f6b873 Mon Sep 17 00:00:00 2001
From: vraghavulu <venkat.raghavulu@intel.com>
Date: Thu, 21 Sep 2023 11:37:54 +0000
Subject: [PATCH] Adding core-avx2 to libc

Signed-off-by: bhavanag <bhavana.g@intel.com>
---
 libc/Android.bp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/libc/Android.bp b/libc/Android.bp
index e32f10fe9..34e030b53 100644
--- a/libc/Android.bp
+++ b/libc/Android.bp
@@ -76,6 +76,9 @@ libc_common_flags = [
     // ever touch 0, 1, or 2 bytes into a call to memset, which was never going
     // to amortize.)
     "-fno-builtin",
+
+    // Adding core-avx2 flag
+    "-march=core-avx2",
 ]
 
 // Define some common cflags
-- 
2.39.2

