From 4090f13cbd5ae0cab81233dd850ce7a8d9fcf660 Mon Sep 17 00:00:00 2001
From: "G, Bhavana" <bhavana.g@intel.com>
Date: Tue, 10 Oct 2023 14:08:41 +0530
Subject: [PATCH] Enable core-avx2 for libc

Signed-off-by: G, Bhavana <bhavana.g@intel.com>
---
 libc/Android.bp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/libc/Android.bp b/libc/Android.bp
index e53ed5f65..e633eac3b 100644
--- a/libc/Android.bp
+++ b/libc/Android.bp
@@ -68,6 +68,9 @@ libc_common_flags = [
 
     // GWP-ASan requires platform TLS.
     "-fno-emulated-tls",
+    
+    // Adding core-avx2 flag
+    "-march=core-avx2",
 ]
 
 // Define some common cflags
-- 
2.34.1

