From 6f920e1c556a11ca5b85963a19fee3def365d0a3 Mon Sep 17 00:00:00 2001
From: "Wang, ArvinX" <arvinx.wang@intel.com>
Date: Fri, 29 Nov 2019 10:19:14 +0800
Subject: [PATCH] change PRODUCT_PACKAGES

---
 target/product/core.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/target/product/core.mk b/target/product/core.mk
index c72bb2f26..d7a4ede64 100644
--- a/target/product/core.mk
+++ b/target/product/core.mk
@@ -42,7 +42,7 @@ PRODUCT_PACKAGES += \
     KeyChain \
     Keyguard \
     LatinIME \
-    Launcher3QuickStep \
+    ShellyTestLauncher \
     ManagedProvisioning \
     MtpDocumentsProvider \
     PacProcessor \
-- 
2.17.1

