From a7856b19fcd730b7756fb2c516289536769199e8 Mon Sep 17 00:00:00 2001
From: "Chen, Yu" <yu.y.chen@intel.com>
Date: Fri, 9 Oct 2020 19:27:59 +0800
Subject: [PATCH] hack selinux of increment fs

---
 incfs/incfs.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/incfs/incfs.cpp b/incfs/incfs.cpp
index 7994c54..cc09003 100644
--- a/incfs/incfs.cpp
+++ b/incfs/incfs.cpp
@@ -457,7 +457,11 @@ IncFsControl* IncFs_Mount(const char* backingPath, const char* targetDir,
     }
 
     if (!restoreconControlFiles(targetDir)) {
-        return nullptr;
+	    auto control = makeControl(targetDir);
+	    if (control == nullptr) {
+		return nullptr;
+	    }
+	    return control;
     }
 
     auto control = makeControl(targetDir);
-- 
2.28.0

