From 6580975a12f418735f07914c93d2889b06d7b749 Mon Sep 17 00:00:00 2001
From: "Chen, Yu" <yu.y.chen@intel.com>
Date: Fri, 9 Oct 2020 19:27:59 +0800
Subject: [PATCH] hack selinux of increment fs

---
 incfs/incfs.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/incfs/incfs.cpp b/incfs/incfs.cpp
index 7994c54..b63080e 100644
--- a/incfs/incfs.cpp
+++ b/incfs/incfs.cpp
@@ -456,9 +456,11 @@ IncFsControl* IncFs_Mount(const char* backingPath, const char* targetDir,
         return nullptr;
     }
 
+/*
     if (!restoreconControlFiles(targetDir)) {
         return nullptr;
     }
+*/
 
     auto control = makeControl(targetDir);
     if (control == nullptr) {
-- 
2.21.0

