From c1803abfe3f59d91b1904a31a62e1cf4dd319eaf Mon Sep 17 00:00:00 2001
From: "Chen, Yu" <yu.y.chen@intel.com>
Date: Sat, 10 Oct 2020 12:47:35 +0800
Subject: [PATCH] recure

---
 incfs/incfs.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/incfs/incfs.cpp b/incfs/incfs.cpp
index 7994c54..8cb3ceb 100644
--- a/incfs/incfs.cpp
+++ b/incfs/incfs.cpp
@@ -408,7 +408,7 @@ static bool restoreconControlFiles(std::string_view targetDir) {
                                             path::join(targetDir, INCFS_LOG_FILENAME)};
     for (size_t i = 0; i < std::size(controlFilePaths); i++) {
         if (const auto err = selinux_android_restorecon(controlFilePaths[i].c_str(),
-                                                        SELINUX_ANDROID_RESTORECON_FORCE);
+                                                        SELINUX_ANDROID_RESTORECON_RECURSE);
             err != 0) {
             PLOG(ERROR) << "[incfs] Failed to restorecon: " << controlFilePaths[i]
                         << " error code: " << err;
-- 
2.28.0

