From 06f3dce1f4add38b98dc605e2bc7549db7c3907d Mon Sep 17 00:00:00 2001
From: anuvarsh <anuvarshini.bc@intel.com>
Date: Tue, 4 Jun 2019 11:45:30 +0530
Subject: [PATCH] ART haswell removed

Tracked-On: None
Signed-off-by: anuvarsh <anuvarshini.bc@intel.com>
---
 build/Android.bp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/build/Android.bp b/build/Android.bp
index 863038eed..e00262f3f 100644
--- a/build/Android.bp
+++ b/build/Android.bp
@@ -77,7 +77,6 @@ art_global_defaults {
                 // "-fno-omit-frame-pointer",
                 // "-marm",
                 // "-mapcs",
-                   "-march=haswell",
             ],
             include_dirs: [
                 // We optimize Thread::Current() with a direct TLS access. This requires access to a
-- 
2.21.0

