From d17294efd86096f238f256a920b21b2f5ff58e0f Mon Sep 17 00:00:00 2001
From: anuvarsh <anuvarshini.bc@intel.com>
Date: Tue, 4 Jun 2019 13:08:47 +0530
Subject: [PATCH] Changed implementation of haswell ART

Tracked-On: None
Signed-off-by: anuvarsh <anuvarshini.bc@intel.com>
---
 build/Android.bp | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/build/Android.bp b/build/Android.bp
index e00262f3f..36653ebf3 100644
--- a/build/Android.bp
+++ b/build/Android.bp
@@ -78,6 +78,19 @@ art_global_defaults {
                 // "-marm",
                 // "-mapcs",
             ],
+            
+            arch: {
+      	       x86: {
+                  avx2: {
+                     cflags: ["-mavx","-mavx2","-mfma"],
+                 },
+              },
+              x86_64: {
+                   avx2: {
+                      cflags: ["-mavx","-mavx2","-mfma"],
+                  },
+                },
+             }, 
             include_dirs: [
                 // We optimize Thread::Current() with a direct TLS access. This requires access to a
                 //  private Bionic header.
-- 
2.21.0

