From 071f41d931a4696df2775b66eebdc74d17d81cd1 Mon Sep 17 00:00:00 2001
From: ahs <amrita.h.s@intel.com>
Date: Tue, 17 Mar 2020 18:23:30 +0530
Subject: Add host support for libdrm

Tracked-On:
Signed-off-by: ahs <amrita.h.s@intel.com>
---
 Android.bp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Android.bp b/Android.bp
index 6fe434c6..5d52668c 100644
--- a/Android.bp
+++ b/Android.bp
@@ -50,6 +50,7 @@ cc_library {
     name: "libdrm",
     recovery_available: true,
     vendor_available: true,
+    host_supported: true,
     defaults: [
         "libdrm_defaults",
         "libdrm_sources",
-- 
2.24.0

