From 815e6ce3600c17abcb87947e633fb90d8fd2933c Mon Sep 17 00:00:00 2001
From: anuvarsh <anuvarshini.bc@intel.com>
Date: Mon, 21 Oct 2019 15:48:21 +0530
Subject: [PATCH] lipjepg try1

Tracked-On:
Signed-off-by: anuvarsh <anuvarshini.bc@intel.com>
---
 Android.bp | 17 -----------------
 1 file changed, 17 deletions(-)

diff --git a/Android.bp b/Android.bp
index e3454fd..db38644 100644
--- a/Android.bp
+++ b/Android.bp
@@ -144,31 +144,14 @@ cc_defaults {
             // x86-64 SSE2
             srcs: [
                 "simd/x86_64/jccolor-avx2.asm",
-                "simd/x86_64/jccolor-sse2.asm",
                 "simd/x86_64/jcgray-avx2.asm",
-                "simd/x86_64/jcgray-sse2.asm",
-                "simd/x86_64/jchuff-sse2.asm",
-                "simd/x86_64/jcphuff-sse2.asm",
                 "simd/x86_64/jcsample-avx2.asm",
-                "simd/x86_64/jcsample-sse2.asm",
                 "simd/x86_64/jdcolor-avx2.asm",
-                "simd/x86_64/jdcolor-sse2.asm",
                 "simd/x86_64/jdmerge-avx2.asm",
-                "simd/x86_64/jdmerge-sse2.asm",
                 "simd/x86_64/jdsample-avx2.asm",
-                "simd/x86_64/jdsample-sse2.asm",
-                "simd/x86_64/jfdctflt-sse.asm",
-                "simd/x86_64/jfdctfst-sse2.asm",
                 "simd/x86_64/jfdctint-avx2.asm",
-                "simd/x86_64/jfdctint-sse2.asm",
-                "simd/x86_64/jidctflt-sse2.asm",
-                "simd/x86_64/jidctfst-sse2.asm",
                 "simd/x86_64/jidctint-avx2.asm",
-                "simd/x86_64/jidctint-sse2.asm",
-                "simd/x86_64/jidctred-sse2.asm",
-                "simd/x86_64/jquantf-sse2.asm",
                 "simd/x86_64/jquanti-avx2.asm",
-                "simd/x86_64/jquanti-sse2.asm",
                 "simd/x86_64/jsimd.c",
                 "simd/x86_64/jsimdcpu.asm",
             ],
-- 
2.23.0

