From b2e86e6ad8f02f93979e89f388a84931ae4547d5 Mon Sep 17 00:00:00 2001
From: "Soni, Ravi Kumar" <ravi.kumar.soni@intel.com>
Date: Wed, 28 Dec 2022 11:08:55 +0530
Subject: [PATCH] arm-optimized-routines libm avx2 optimization

Signed-off-by: Soni, Ravi Kumar <ravi.kumar.soni@intel.com>
---
 Android.bp  | 19 +++++++++++++++++++
 math/log.c  |  2 ++
 math/logf.c |  2 ++
 3 files changed, 23 insertions(+)

diff --git a/Android.bp b/Android.bp
index ba814eb..efd97b2 100755
--- a/Android.bp
+++ b/Android.bp
@@ -85,6 +85,25 @@ cc_library_static {
                 "-DHAVE_FAST_FMA=1",
             ],
         },
+        x86: {
+           exclude_srcs: [
+               "math/sinf.c",
+               "math/cosf.c",
+               "math/sincosf.c",
+               "math/log.c",
+               "math/logf.c",
+           ],
+        },
+	x86_64: {
+           exclude_srcs: [
+               "math/sinf.c",
+               "math/cosf.c",
+	       "math/sincosf.c",
+               "math/log.c",
+	       "math/logf.c",
+           ],
+
+	},
     },
 
     target: {
diff --git a/math/log.c b/math/log.c
index d3b7bc6..c964954 100644
--- a/math/log.c
+++ b/math/log.c
@@ -154,8 +154,10 @@ log (double x)
   return eval_as_double (y);
 }
 #if USE_GLIBC_ABI
+#ifndef LOG_FROM_LIBM
 strong_alias (log, __log_finite)
 hidden_alias (log, __ieee754_log)
+#endif
 # if LDBL_MANT_DIG == 53
 long double logl (long double x) { return log (x); }
 # endif
diff --git a/math/logf.c b/math/logf.c
index cfbaee1..e7f7c70 100644
--- a/math/logf.c
+++ b/math/logf.c
@@ -74,6 +74,8 @@ logf (float x)
   return eval_as_float (y);
 }
 #if USE_GLIBC_ABI
+#ifndef LOGF_FROM_LIBM
 strong_alias (logf, __logf_finite)
 hidden_alias (logf, __ieee754_logf)
 #endif
+#endif
-- 
2.25.1

