From 16826a59d7790c7c467cf7cbb4d93adb8b299d11 Mon Sep 17 00:00:00 2001
From: Shwetha B <shwetha.b@intel.com>
Date: Mon, 17 Jul 2023 14:32:03 +0530
Subject: [PATCH] Change march to alderlake

Signed-off-by: Shwetha B <shwetha.b@intel.com>
---
 cc/config/x86_64_device.go | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/cc/config/x86_64_device.go b/cc/config/x86_64_device.go
index e4c3c7fe2..c8cc6c47f 100644
--- a/cc/config/x86_64_device.go
+++ b/cc/config/x86_64_device.go
@@ -36,7 +36,7 @@ var (
 
 	x86_64ArchVariantCflags = map[string][]string{
 		"": []string{
-			"-march=x86-64",
+			"-march=alderlake",
 		},
 		"broadwell": []string{
 			"-march=broadwell",
@@ -52,7 +52,7 @@ var (
 			"-march=corei7",
 		},
 		"silvermont": []string{
-			"-march=slm",
+			"-march=alderlake",
 		},
 		"skylake": []string{
 		},
-- 
2.40.1

