From e7d424443ebd5db08d9a54280d9a995be363e410 Mon Sep 17 00:00:00 2001
From: Shwetha B <shwetha.b@intel.com>
Date: Mon, 17 Jul 2023 15:36:14 +0530
Subject: [PATCH] [Not merge] Change mtrace to core-avx2

Signed-off-by: Shwetha B <shwetha.b@intel.com>
---
 cc/config/x86_64_device.go | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/cc/config/x86_64_device.go b/cc/config/x86_64_device.go
index e4c3c7fe2..420959bcd 100644
--- a/cc/config/x86_64_device.go
+++ b/cc/config/x86_64_device.go
@@ -36,7 +36,7 @@ var (
 
 	x86_64ArchVariantCflags = map[string][]string{
 		"": []string{
-			"-march=x86-64",
+			"-march=core-avx2",
 		},
 		"broadwell": []string{
 			"-march=broadwell",
@@ -52,7 +52,7 @@ var (
 			"-march=corei7",
 		},
 		"silvermont": []string{
-			"-march=slm",
+			"-march=core-avx2",
 		},
 		"skylake": []string{
 		},
-- 
2.40.1

