From d8ee9358dfe24012a1145dae4f2360853358b1e0 Mon Sep 17 00:00:00 2001
From: Swaroop Balan <swaroop.balan@intel.com>
Date: Fri, 17 Aug 2018 09:25:11 +0530
Subject: [PATCH] [CELADON_P] Disabling for user build

Tracked-On: None
---
 Android.mk | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/Android.mk b/Android.mk
index a6c3f31..c323406 100644
--- a/Android.mk
+++ b/Android.mk
@@ -83,9 +83,6 @@ endif
 SELINUX_IGNORE_NEVERALLOWS := true
 NEVERALLOW_ARG :=
 ifeq ($(SELINUX_IGNORE_NEVERALLOWS),true)
-ifeq ($(TARGET_BUILD_VARIANT),user)
-$(error SELINUX_IGNORE_NEVERALLOWS := true cannot be used in user builds)
-endif
 $(warning Be careful when using the SELINUX_IGNORE_NEVERALLOWS flag. \
           It does not work in user builds and using it will \
           not stop you from failing CTS.)
-- 
1.9.1

