From 89e2e9c49423418bb145cecc9c13d44b4ebd1add Mon Sep 17 00:00:00 2001
From: Jingdong Lu <jingdong.lu@intel.com>
Date: Tue, 15 Aug 2023 16:43:09 +0800
Subject: [PATCH] Support Android integration on x86

Tracked-On: OAM-111851
Signed-off-by: Jingdong Lu <jingdong.lu@intel.com>
---
 Android.mk | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/Android.mk b/Android.mk
index 28ccce8..ed437b8 100644
--- a/Android.mk
+++ b/Android.mk
@@ -9,8 +9,8 @@ CFG_TEE_SUPP_LOG_LEVEL ?= 2
 
 # Define Android-specific configuration before including config.mk
 CFG_TEE_CLIENT_LOAD_PATH ?= /vendor/lib
-TEEC_TEST_LOAD_PATH ?= /data/vendor/tee
-CFG_TEE_FS_PARENT_PATH ?= /data/vendor/tee
+TEEC_TEST_LOAD_PATH ?= /persist/tee
+CFG_TEE_FS_PARENT_PATH ?= /persist/tee
 ifneq ($(strip $($(combo_2nd_arch_prefix)TARGET_TOOLS_PREFIX)),)
 CFG_TEE_PLUGIN_LOAD_PATH ?= /vendor/lib64/tee-supplicant/plugins/
 else
@@ -59,7 +59,7 @@ LOCAL_VENDOR_MODULE := true
 
 # Build the 32-bit and 64-bit versions.
 LOCAL_MULTILIB := both
-LOCAL_MODULE_TARGET_ARCH := arm arm64
+LOCAL_MODULE_TARGET_ARCH := x86 x86_64
 
 LOCAL_EXPORT_C_INCLUDE_DIRS := $(LOCAL_PATH)/public
 
@@ -94,7 +94,7 @@ LOCAL_VENDOR_MODULE := true
 
 # Build the 32-bit and 64-bit versions.
 LOCAL_MULTILIB := both
-LOCAL_MODULE_TARGET_ARCH := arm arm64
+LOCAL_MODULE_TARGET_ARCH := x86 x86_64
 
 LOCAL_EXPORT_C_INCLUDE_DIRS := $(LOCAL_PATH)/libckteec/include
 
-- 
2.17.1

