From b5dbcff834600c9a8c3c41f0972d98b7ca34d23c Mon Sep 17 00:00:00 2001
From: "Boppana, Tej Kiran" <tej.kiran.boppana@intel.com>
Date: Thu, 27 Apr 2023 18:53:20 +0530
Subject: [PATCH] DEBUG Patch to enable Kernel Command Line parameters

---
 groups/kernel/BoardConfig.mk | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/groups/kernel/BoardConfig.mk b/groups/kernel/BoardConfig.mk
index 4bb2048..fe882b3 100644
--- a/groups/kernel/BoardConfig.mk
+++ b/groups/kernel/BoardConfig.mk
@@ -71,7 +71,9 @@ endif
 BOARD_KERNEL_CMDLINE += \
       clearcpuid=517 \
       mce=no_lmce \
-      mce=ignore_ce
+      mce=ignore_ce \
+      kmemleak=on \
+      kfence.mode=1
 
 BOARD_SEPOLICY_M4DEFS += module_kernel=true
 BOARD_SEPOLICY_DIRS += $(INTEL_PATH_SEPOLICY)/kernel
-- 
2.17.1

