From 46e9ee61372b174d603631ae048dcec8d1284c0b Mon Sep 17 00:00:00 2001
From: Ratnesh Kumar Rai <ratnesh.kumar.rai@intel.com>
Date: Wed, 13 Sep 2023 21:52:36 +0530
Subject: [PATCH] Disable nnhal

Tracked-On:
Signed-off-by: Ratnesh Kumar Rai <ratnesh.kumar.rai@intel.com>
---
 celadon_ivi/mixins.spec | 1 -
 1 file changed, 1 deletion(-)

diff --git a/celadon_ivi/mixins.spec b/celadon_ivi/mixins.spec
index 17a69d1..00f36a2 100755
--- a/celadon_ivi/mixins.spec
+++ b/celadon_ivi/mixins.spec
@@ -88,6 +88,5 @@ sensors: mediation(enable_sensor_list=true)
 bugreport: true
 mainline-mod: true
 houdini: true
-neuralnetworks: true
 docker: true
 ipu: true
-- 
2.17.1

