From abd589d9e7d3e2cb0146a00f45517041b8243992 Mon Sep 17 00:00:00 2001
From: "Yan, Shaopu" <shaopu.yan@intel.com>
Date: Wed, 10 May 2023 00:02:09 +0800
Subject: [PATCH] remove AOSP initrc for RPMB Mock

Signed-off-by: Yan, Shaopu <shaopu.yan@intel.com>
---
 trusty/utils/rpmb_dev/Android.bp | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/trusty/utils/rpmb_dev/Android.bp b/trusty/utils/rpmb_dev/Android.bp
index a2700876a..ff51445ca 100644
--- a/trusty/utils/rpmb_dev/Android.bp
+++ b/trusty/utils/rpmb_dev/Android.bp
@@ -31,7 +31,4 @@ cc_binary {
         "-Wall",
         "-Werror",
     ],
-    init_rc: [
-        "rpmb_dev.rc",
-    ],
 }
-- 
2.25.1

