From 230693f8e751ccb38a3298a63bcdd71f40fbdfae Mon Sep 17 00:00:00 2001
From: saranya <saranya.gopal@intel.com>
Date: Tue, 20 Oct 2020 23:38:10 +0530
Subject: [PATCH] Don't init usb thread in adb

Signed-off-by: saranya <saranya.gopal@intel.com>
---
 adb/daemon/main.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/adb/daemon/main.cpp b/adb/daemon/main.cpp
index 658e24456..d948b1512 100644
--- a/adb/daemon/main.cpp
+++ b/adb/daemon/main.cpp
@@ -235,7 +235,7 @@ int adbd_main(int server_port) {
 
     bool is_usb = false;
 
-#if defined(__ANDROID__)
+#if 0
     if (access(USB_FFS_ADB_EP0, F_OK) == 0) {
         // Listen on USB.
         usb_init();
-- 
2.17.1

