From 9e26338d466a807af48c09ecf7354ffc85636dd8 Mon Sep 17 00:00:00 2001
From: "Lu, Xingjiang" <xingjiang.lu@intel.com>
Date: Mon, 15 May 2023 14:59:13 +0800
Subject: [PATCH] Add blacklist properties of llkd to avoid the disconnection.

Signed-off-by: Lu, Xingjiang <xingjiang.lu@intel.com>
---
 llkd/llkd.rc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/llkd/llkd.rc b/llkd/llkd.rc
index b1f96a8f3..45dcb6210 100644
--- a/llkd/llkd.rc
+++ b/llkd/llkd.rc
@@ -32,6 +32,8 @@ on property:khungtask.enable=false
     write /proc/sys/kernel/hung_task_panic 0
 
 on property:llk.enable=true
+    setprop ro.llk.blacklist.process adbd
+    setprop ro.llk.blacklist.parent adbd
     start llkd-${ro.debuggable:-0}
 
 service llkd-0 /system/bin/llkd
-- 
2.34.1

