From 25696d5f09a988f8836bdaeae3c4078fda5b6eef Mon Sep 17 00:00:00 2001
From: ahs <amrita.h.s@intel.com>
Date: Tue, 8 Dec 2020 12:20:58 +0530
Subject: [PATCH] Patch avx version of libneuralnetworks with correct soname

Tracked-On:
Signed-off-by: ahs <amrita.h.s@intel.com>
---
 nn/runtime/Android.bp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/nn/runtime/Android.bp b/nn/runtime/Android.bp
index b33358d..554654f 100644
--- a/nn/runtime/Android.bp
+++ b/nn/runtime/Android.bp
@@ -154,6 +154,7 @@ cc_library_shared {
           relative_install_path: "IA-Perf/avx2",
        },
     },
+    ldflags: ["-Wl,-soname,libneuralnetworks.so"],
 }
 
 // Required for tests (b/147158681)
-- 
2.7.4

