From 4ccaba80ab76265c24842f78838d038340ed9c47 Mon Sep 17 00:00:00 2001
From: shivasku82 <shiva.kumara.rudrappa@intel.com>
Date: Tue, 13 Jul 2021 19:27:14 +0530
Subject: [PATCH] buffer pool debug

Signed-off-by: shivasku82 <shiva.kumara.rudrappa@intel.com>
---
 media/bufferpool/2.0/AccessorImpl.cpp | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/media/bufferpool/2.0/AccessorImpl.cpp b/media/bufferpool/2.0/AccessorImpl.cpp
index 6111feafb9..f2cce2e27c 100644
--- a/media/bufferpool/2.0/AccessorImpl.cpp
+++ b/media/bufferpool/2.0/AccessorImpl.cpp
@@ -15,7 +15,7 @@
  */
 
 #define LOG_TAG "BufferPoolAccessor2.0"
-//#define LOG_NDEBUG 0
+#define LOG_NDEBUG 0
 
 #include <sys/types.h>
 #include <stdint.h>
@@ -439,7 +439,7 @@ void Accessor::Impl::BufferPool::Invalidation::onHandleAck(
 
 bool Accessor::Impl::BufferPool::handleOwnBuffer(
         ConnectionId connectionId, BufferId bufferId) {
-
+ALOGE("shiva handleOwnBuffer ");
     bool added = insert(&mUsingBuffers, connectionId, bufferId);
     if (added) {
         auto iter = mBuffers.find(bufferId);
@@ -451,6 +451,8 @@ bool Accessor::Impl::BufferPool::handleOwnBuffer(
 
 bool Accessor::Impl::BufferPool::handleReleaseBuffer(
         ConnectionId connectionId, BufferId bufferId) {
+	ALOGE("shiva handleReleaseBuffer ");
+
     bool deleted = erase(&mUsingBuffers, connectionId, bufferId);
     if (deleted) {
         auto iter = mBuffers.find(bufferId);
@@ -478,6 +480,8 @@ bool Accessor::Impl::BufferPool::handleTransferTo(const BufferStatusMessage &mes
             message.transactionId);
     if (completed != mCompletedTransactions.end()) {
         // already completed
+	        ALOGE("shiva handleTransferTo ");
+
         mCompletedTransactions.erase(completed);
         return true;
     }
@@ -614,6 +618,7 @@ void Accessor::Impl::BufferPool::processStatusMessages() {
 }
 
 bool Accessor::Impl::BufferPool::handleClose(ConnectionId connectionId) {
+	ALOGE("shiva handleClose");
     // Cleaning buffers
     auto buffers = mUsingBuffers.find(connectionId);
     if (buffers != mUsingBuffers.end()) {
@@ -702,7 +707,7 @@ ResultStatus Accessor::Impl::BufferPool::addNewBuffer(
         const std::vector<uint8_t> &params,
         BufferId *pId,
         const native_handle_t** handle) {
-
+ALOGE("shiva addNewBuffer");
     BufferId bufferId = mSeq++;
     if (mSeq == Connection::SYNC_BUFFERID) {
         mSeq = 0;
-- 
2.32.0

