From 114bfc94102c558460009f058a22cd9572951425 Mon Sep 17 00:00:00 2001
From: "Reddy, Alavala Srinivasa" <alavala.srinivasa.reddy@intel.com>
Date: Fri, 10 Nov 2023 15:42:55 +0530
Subject: [PATCH] Modify IA-Perf variants of library to have same name as
 original

Signed-off-by: ahs <amrita.h.s@intel.com>
---
 media/libaudioprocessing/Android.bp | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/media/libaudioprocessing/Android.bp b/media/libaudioprocessing/Android.bp
index 9a5272db64..b5b2177619 100644
--- a/media/libaudioprocessing/Android.bp
+++ b/media/libaudioprocessing/Android.bp
@@ -103,11 +103,8 @@ cc_library_shared {
 
 cc_library_shared {
    name: "libaudioprocessing_avx2",
+   override_lib_name: "libaudioprocessing",
    defaults: ["libaudioprocessing_generic", "libaudioprocessing_defaults_avx2"],
-   target: {
-       android: {
-          relative_install_path: "IA-Perf/avx2",
-       },
-   },
+   relative_install_path: "IA-Perf/avx2",
    whole_static_libs: ["libaudioprocessing_base_avx2"],
 }
-- 
2.17.1

