From 298491ee97803fccb5ce3a9a977871bbe1301f68 Mon Sep 17 00:00:00 2001
From: Shaofeng Tang <shaofeng.tang@intel.com>
Date: Wed, 8 Dec 2021 16:27:24 +0800
Subject: [PATCH] Add log to trace buffer format convertion in SoftwareRender.

Signed-off-by: Shaofeng Tang <shaofeng.tang@intel.com>
---
 media/libstagefright/colorconversion/SoftwareRenderer.cpp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/media/libstagefright/colorconversion/SoftwareRenderer.cpp b/media/libstagefright/colorconversion/SoftwareRenderer.cpp
index 4711315615..e1fa121de4 100644
--- a/media/libstagefright/colorconversion/SoftwareRenderer.cpp
+++ b/media/libstagefright/colorconversion/SoftwareRenderer.cpp
@@ -107,10 +107,14 @@ void SoftwareRenderer::resetFormatIfChanged(
         mCropBottom == cropBottomNew &&
         !dataSpaceChangedForPlanar16) {
         // Nothing changed, no need to reset renderer.
+	ALOGE("####### SoftwareRenderer init color format with %#x", mColorFormat);
         return;
     }
 
     mColorFormat = static_cast<OMX_COLOR_FORMATTYPE>(colorFormatNew);
+
+    ALOGE("####### SoftwareRenderer reset color format with %#x", mColorFormat);
+
     mWidth = widthNew;
     mHeight = heightNew;
     mStride = strideNew;
@@ -175,16 +179,19 @@ void SoftwareRenderer::resetFormatIfChanged(
                 break;
             }
         }
+	ALOGE("####### SoftwareRenderer HAL color format is %#x", halFormat);
     }
 
     if (halFormat == HAL_PIXEL_FORMAT_RGB_565) {
         mConverter = new ColorConverter(
                 mColorFormat, OMX_COLOR_Format16bitRGB565);
         CHECK(mConverter->isValid());
+        ALOGE("####### SoftwareRenderer create ColorConverter to convert mColorFormat %#x to OMX_COLOR_Format16bitRGB565 for HAL_PIXEL_FORMAT_RGB_565", mColorFormat);
     } else if (halFormat == HAL_PIXEL_FORMAT_RGBA_1010102) {
         mConverter = new ColorConverter(
                 mColorFormat, OMX_COLOR_FormatYUV444Y410);
         CHECK(mConverter->isValid());
+	ALOGE("####### SoftwareRenderer create ColorConverter to convert mColorFormat %#x to OMX_COLOR_FormatYUV444Y410 for HAL_PIXEL_FORMAT_RGBA_1010102", mColorFormat);
     }
 
     CHECK(mNativeWindow != NULL);
-- 
2.29.2

