From f44aa89a598bb97fd0666445435019fba38da964 Mon Sep 17 00:00:00 2001
From: ahs <amrita.h.s@intel.com>
Date: Tue, 8 Dec 2020 12:38:21 +0530
Subject: [PATCH] Patch avx version of libaudioprocessing with correct soname

Tracked-On:
Signed-off-by: ahs <amrita.h.s@intel.com>
---
 media/libaudioprocessing/Android.bp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/media/libaudioprocessing/Android.bp b/media/libaudioprocessing/Android.bp
index 0b01fde..8315b68 100644
--- a/media/libaudioprocessing/Android.bp
+++ b/media/libaudioprocessing/Android.bp
@@ -101,4 +101,5 @@ cc_library_shared {
        },
    },
    whole_static_libs: ["libaudioprocessing_base_avx2"],
+   ldflags: ["-Wl,-soname,libaudioprocessing.so"],
 }
-- 
2.7.4

