From 59082ed198fc3dc9f57a0b70fcfe612bcb906ce0 Mon Sep 17 00:00:00 2001
From: "Wan, Hao" <haox.wan@intel.com>
Date: Mon, 15 Apr 2024 17:36:24 +0000
Subject: [PATCH] Disable sw codec property check

Signed-off-by: Wan, Hao <haox.wan@intel.com>
---
 WrsOMXPlugin.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/WrsOMXPlugin.cpp b/WrsOMXPlugin.cpp
index d4afcf0..44b70d7 100644
--- a/WrsOMXPlugin.cpp
+++ b/WrsOMXPlugin.cpp
@@ -24,6 +24,8 @@
 #include <HardwareAPI.h>
 #include <media/stagefright/foundation/ADebug.h>
 
+#define DISABLE_SW_CODEC 0
+
 namespace android {
 
 OMXPluginBase *createOMXPlugin() {
@@ -55,12 +57,14 @@ OMX_ERRORTYPE WrsOMXPlugin::AddCore(const char* coreName)
 
    char szVendorIntelVideoCodec[PROPERTY_VALUE_MAX] = {'\0'};
 
+#if DISABLE_SW_CODEC
    if(property_get("vendor.intel.video.codec", szVendorIntelVideoCodec, NULL) > 0 ) {
        if (strncmp(szVendorIntelVideoCodec, "software", PROPERTY_VALUE_MAX) == 0 ) {
            ALOGI("OMX IL core Property vendor.intel.video.codec is %s and will not load hardware codec plugin", szVendorIntelVideoCodec);
            return OMX_ErrorNone;
        }
    }
+#endif
 
    if (libHandle != NULL) {
         WrsOMXCore* core = (WrsOMXCore*)calloc(1,sizeof(WrsOMXCore));
-- 
2.34.1

