From c5059ef27f1a6a0a795f42c587de050716fff237 Mon Sep 17 00:00:00 2001
From: Kalyan Kondapally <kalyan.kondapally@intel.com>
Date: Sun, 17 Jul 2016 20:16:12 -0700
Subject: [PATCH 1/4] Adopt to latest libsync API changes.

---
 libs/ui/Fence.cpp | 15 +++++----------
 1 file changed, 5 insertions(+), 10 deletions(-)

diff --git a/libs/ui/Fence.cpp b/libs/ui/Fence.cpp
index bf24ffb..fb789c3 100644
--- a/libs/ui/Fence.cpp
+++ b/libs/ui/Fence.cpp
@@ -107,24 +107,19 @@ nsecs_t Fence::getSignalTime() const {
         return -1;
     }

-    struct sync_fence_info_data* finfo = sync_fence_info(mFenceFd);
+    struct sync_file_info* finfo = sync_file_info(mFenceFd);
     if (finfo == NULL) {
         ALOGE("sync_fence_info returned NULL for fd %d", mFenceFd);
         return -1;
     }
     if (finfo->status != 1) {
-        sync_fence_info_free(finfo);
+        sync_file_info_free(finfo);
         return INT64_MAX;
     }

-    struct sync_pt_info* pinfo = NULL;
-    uint64_t timestamp = 0;
-    while ((pinfo = sync_pt_info(finfo, pinfo)) != NULL) {
-        if (pinfo->timestamp_ns > timestamp) {
-            timestamp = pinfo->timestamp_ns;
-        }
-    }
-    sync_fence_info_free(finfo);
+    uint64_t timestamp = sync_fence_timestamp(finfo);
+
+    sync_file_info_free(finfo);

     return nsecs_t(timestamp);
 }
--
2.7.4
