From 7ef1512d0469b3558f87d4bc66c3220ab3d0619e Mon Sep 17 00:00:00 2001
From: "Zhang, Qi" <qi1.zhang@intel.com>
Date: Mon, 23 Sep 2019 09:42:49 +0800
Subject: [PATCH] enable vt-d for aiv

Signed-off-by: Zhang, Qi <qi1.zhang@intel.com>
---
 product/celadon_64.cfg | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/product/celadon_64.cfg b/product/celadon_64.cfg
index caab540..c10c134 100644
--- a/product/celadon_64.cfg
+++ b/product/celadon_64.cfg
@@ -17,10 +17,10 @@ EVMM_CMPL_FLAGS += \
  -DPACK_LK \
  -DMODULE_ACPI
 
-#EVMM_CMPL_FLAGS += \
-# -DMODULE_VTD \
-# -DDMAR_MAX_ENGINE=16 \
-# -DSKIP_DMAR_GPU
+EVMM_CMPL_FLAGS += \
+ -DMODULE_VTD \
+ -DDMAR_MAX_ENGINE=16 \
+ -DSKIP_DMAR_GPU
 
 EVMM_CMPL_FLAGS += \
  -DAP_START_IN_HLT
-- 
2.20.1

