From ce1af09b775c1a429db4a634d893f3ca3ec96bf4 Mon Sep 17 00:00:00 2001
From: Raveendra Babu Chennakesavulu <raveendra.babu.chennakesavulu@intel.com>
Date: Wed, 20 May 2020 23:11:15 +0530
Subject: [PATCH] [packages/apps] Ethernet proxy via WiFi Settings

Change-Id: Ia02c623c079c7df392e64ada734943342c330ae5
Tracked-On:

diff --git a/src/com/android/settings/wifi/WifiSettings.java b/src/com/android/settings/wifi/WifiSettings.java
index 7a7caf47ef..b2376b3c87 100644
--- a/src/com/android/settings/wifi/WifiSettings.java
+++ b/src/com/android/settings/wifi/WifiSettings.java
@@ -33,6 +33,10 @@ import android.net.NetworkInfo.State;
 import android.net.NetworkRequest;
 import android.net.wifi.WifiConfiguration;
 import android.net.wifi.WifiManager;
+import android.net.EthernetManager;
+import android.net.IpConfiguration;
+import android.net.ProxyInfo;
+import android.net.IpConfiguration.ProxySettings;
 import android.nfc.NfcAdapter;
 import android.os.Bundle;
 import android.os.Handler;
@@ -119,6 +123,9 @@ public class WifiSettings extends RestrictedSettingsFragment
         setProgressBarVisible(false);
     };
 
+    //Raveendra
+    private EthernetManager mEthernetManager;
+    private IpConfiguration mIpConfiguration;
     protected WifiManager mWifiManager;
     private ConnectivityManager mConnectivityManager;
     private WifiManager.ActionListener mConnectListener;
@@ -1020,6 +1027,8 @@ public class WifiSettings extends RestrictedSettingsFragment
     /* package */ void submit(WifiConfigController configController) {
 
         final WifiConfiguration config = configController.getConfig();
+	final Context mContext = getContext();
+	final String mInterfaceName;
 
         if (config == null) {
             if (mSelectedAccessPoint != null
@@ -1034,6 +1043,21 @@ public class WifiSettings extends RestrictedSettingsFragment
                 connect(config, false /* isSavedNetwork */);
             }
         }
+	mEthernetManager = (EthernetManager) mContext.getSystemService(Context.ETHERNET_SERVICE);
+	String[] ifaces = mEthernetManager.getAvailableInterfaces();
+        if (ifaces.length > 0) {
+            mInterfaceName = ifaces[0];
+	    Log.d(TAG, "Raveendra: submit, mInterfaceName: " + mInterfaceName);
+            mIpConfiguration = mEthernetManager.getConfiguration(mInterfaceName);
+
+	    mIpConfiguration.setProxySettings(ProxySettings.STATIC);
+	    ProxyInfo mHttpProxy = new ProxyInfo(config.getHttpProxy());
+	    mIpConfiguration.setHttpProxy(mHttpProxy);
+	    Log.d(TAG, "Raveendra: submit, mHttpProxy: " + mHttpProxy);
+	    if (mInterfaceName != null) {
+		mEthernetManager.setConfiguration(mInterfaceName, mIpConfiguration);
+	    }
+        }
 
         mWifiTracker.resumeScanning();
     }
-- 
2.17.1

