From 024a373cac2c360b32ecd094ccb6b85c5fe55516 Mon Sep 17 00:00:00 2001
From: rajucm <raju.mallikarjun.chegaraddi@intel.com>
Date: Tue, 22 Aug 2023 04:38:28 +0000
Subject: [PATCH] gpt_image: add swap partition

Signed-off-by: rajucm <raju.mallikarjun.chegaraddi@intel.com>
---
 create_gpt_image.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/create_gpt_image.py b/create_gpt_image.py
index 16ad850..95fd8dc 100755
--- a/create_gpt_image.py
+++ b/create_gpt_image.py
@@ -517,6 +517,8 @@ class TableEntryInfos(object):
                                           '3d69d8477de4'),
                       'config': ('0fc63daf-8483-4772-8e79-'
                                          '3d69d8477de4'),
+                      'swap': ('0fc63daf-8483-4772-8e79-'
+                                         '3d69d8477de4'),
                       'mfos':'4f33cfe4-a0c1-448b-aec4-40f10a0cef3f',
                       'teedata': '0fc63daf-8483-4772-8e79-3d69d8477de4',
                       'super': '0fc63daf-8483-4772-8e79-3d69d8477de4',
@@ -872,6 +874,7 @@ class GPTImage(object):
         'persistent',
         'factory',
         'config',
+        'swap',
         'tos',
         'mfos',
         'teedata',
-- 
2.39.2

