From 352b74bf3ed8f51839ab3911f775b9eb32d0f0ad Mon Sep 17 00:00:00 2001
From: Meng Xianglin <xianglinx.meng@intel.com>
Date: Tue, 24 Sep 2019 13:04:47 +0800
Subject: [PATCH] Fix an issue can cause NULL pointer reference

Tracked-On: OAM-86789
Signed-off-by: Meng Xianglin <xianglinx.meng@intel.com>
---
 libkernelflinger/UsbMassBot.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/libkernelflinger/UsbMassBot.c b/libkernelflinger/UsbMassBot.c
index 82fabf0..cf38c19 100644
--- a/libkernelflinger/UsbMassBot.c
+++ b/libkernelflinger/UsbMassBot.c
@@ -82,7 +82,9 @@ UsbBotInit (
   // Allocate the BOT context for USB_BOT_PROTOCOL and two endpoint descriptors.
   //
   UsbBot = AllocateZeroPool (sizeof (USB_BOT_PROTOCOL) + 2 * sizeof (EFI_USB_ENDPOINT_DESCRIPTOR));
-  ASSERT (UsbBot != NULL);
+  if (UsbBot == NULL)
+    return EFI_OUT_OF_RESOURCES;
+
 
   UsbBot->UsbIo = UsbIo;
 
