From c08cb012b0dbad680f98209b8429dbee19133582 Mon Sep 17 00:00:00 2001
From: Ming Tan <ming.tan@intel.com>
Date: Tue, 25 Dec 2018 10:19:50 +0800
Subject: [PATCH] Set ro.zygote=zygote64_32.

Signed-off-by: Ming Tan <ming.tan@intel.com>
---
 cel_apl/cel_apl.mk | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/cel_apl/cel_apl.mk b/cel_apl/cel_apl.mk
index a98e9ea..34e9f83 100644
--- a/cel_apl/cel_apl.mk
+++ b/cel_apl/cel_apl.mk
@@ -1,6 +1,9 @@
 include build/make/target/product/treble_common.mk
 BOARD_PROPERTY_OVERRIDES_SPLIT_ENABLED := true
 
+PRODUCT_DEFAULT_PROPERTY_OVERRIDES += ro.zygote=zygote64_32
+PRODUCT_COPY_FILES += system/core/rootdir/init.zygote64_32.rc:root/init.zygote64_32.rc
+
 ENABLE_NATIVEBRIDGE_64BIT := true
 
 $(call inherit-product,$(SRC_TARGET_DIR)/product/core_64_bit.mk)
-- 
1.9.1

