From 1cfc616118ffc0ac98cdbb6e3413c613345cd2f4 Mon Sep 17 00:00:00 2001
From: Sun Jian <jian.sun@intel.com>
Date: Mon, 4 Oct 2021 06:41:16 +0800
Subject: [PATCH] Skip hw pmu for VM on ADL

---
 arch/x86/events/core.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/arch/x86/events/core.c b/arch/x86/events/core.c
index 81d5e0a1f48c..0594235a0cd0 100644
--- a/arch/x86/events/core.c
+++ b/arch/x86/events/core.c
@@ -41,6 +41,7 @@
 #include <asm/desc.h>
 #include <asm/ldt.h>
 #include <asm/unwind.h>
+#include <asm/intel-family.h>
 
 #include "perf_event.h"
 
@@ -252,6 +253,13 @@ bool check_hw_exists(struct pmu *pmu, int num_counters, int num_counters_fixed)
 	int bios_fail = 0;
 	int reg_safe = -1;
 
+	if(boot_cpu_has(X86_FEATURE_HYPERVISOR) &&
+			(boot_cpu_data.x86_model == INTEL_FAM6_ALDERLAKE ||
+			boot_cpu_data.x86_model == INTEL_FAM6_ALDERLAKE_L)) {
+		pr_cont("Disabling PMU for VM on ADL.\n");
+		goto msr_fail;
+	}
+
 	/*
 	 * Check to see if the BIOS enabled any of the counters, if so
 	 * complain and bail.
-- 
2.35.1

