From 400e7c93ac31484471b6f354a546cd274df3e328 Mon Sep 17 00:00:00 2001
From: HeYue <yue.he@intel.com>
Date: Mon, 11 Sep 2023 09:18:38 +0800
Subject: [PATCH] Disable ADL&RPL iGPU in kernel

Signed-off-by: HeYue <yue.he@intel.com>
---
 include/drm/i915_pciids.h | 2 --
 1 file changed, 2 deletions(-)

diff --git a/include/drm/i915_pciids.h b/include/drm/i915_pciids.h
index 4a4c190f7698..ff295370d948 100644
--- a/include/drm/i915_pciids.h
+++ b/include/drm/i915_pciids.h
@@ -652,7 +652,6 @@
 	INTEL_VGA_DEVICE(0x46A1, info), \
 	INTEL_VGA_DEVICE(0x46A2, info), \
 	INTEL_VGA_DEVICE(0x46A3, info), \
-	INTEL_VGA_DEVICE(0x46A6, info), \
 	INTEL_VGA_DEVICE(0x46A8, info), \
 	INTEL_VGA_DEVICE(0x46AA, info), \
 	INTEL_VGA_DEVICE(0x462A, info), \
@@ -688,7 +687,6 @@
 #define INTEL_RPLP_IDS(info) \
 	INTEL_VGA_DEVICE(0xA720, info), \
 	INTEL_VGA_DEVICE(0xA721, info), \
-	INTEL_VGA_DEVICE(0xA7A0, info), \
 	INTEL_VGA_DEVICE(0xA7A1, info), \
 	INTEL_VGA_DEVICE(0xA7A8, info), \
 	INTEL_VGA_DEVICE(0xA7A9, info)
-- 
2.17.1

