From a364c06ab1f9c11ac27e3a4e7ac83b53924c38a6 Mon Sep 17 00:00:00 2001
From: "Xue, Bosheng" <bosheng.xue@intel.com>
Date: Wed, 8 Nov 2023 11:06:01 +0800
Subject: [PATCH 4/4] add DRM_PLANE_NO_SCALING for build failure

---
 include/drm/drm_plane.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/drm/drm_plane.h b/include/drm/drm_plane.h
index 8989bb1aa46c..8bea86542820 100644
--- a/include/drm/drm_plane.h
+++ b/include/drm/drm_plane.h
@@ -617,6 +617,8 @@ enum drm_plane_type {
 	 * support modifiers, the framebuffer should have a linear layout.
 	 */
 	DRM_PLANE_TYPE_CURSOR,
+
+	DRM_PLANE_NO_SCALING,
 };
 
 
-- 
2.34.1

