From ea8a2c9381e938c10e58ac9a73b7812a5c12d490 Mon Sep 17 00:00:00 2001
From: anuvarsh <anuvarshini.bc@intel.com>
Date: Mon, 14 Oct 2019 14:24:03 +0530
Subject: [PATCH] HWUI

Tracked-On:
Signed-off-by: anuvarsh <anuvarshini.bc@intel.com>
---
 libs/hwui/Android.bp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libs/hwui/Android.bp b/libs/hwui/Android.bp
index cb260b5b4b9..f52997e7742 100644
--- a/libs/hwui/Android.bp
+++ b/libs/hwui/Android.bp
@@ -21,6 +21,8 @@ cc_defaults {
         "-Wunreachable-code",
         "-Werror",
         "-fvisibility=hidden",
+	"-mavx2",
+	"-mfma",
 
         // GCC false-positives on this warning, and since we -Werror that's
         // a problem
-- 
2.23.0

