From 9afd5263543b5da19883a63430af39f98f812aa9 Mon Sep 17 00:00:00 2001
From: Salini Venate <salini.venate@intel.com>
Date: Fri, 21 Jun 2024 05:58:55 +0000
Subject: [PATCH] disable trusty

---
 caas/mixins.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/caas/mixins.spec b/caas/mixins.spec
index d8b1494..8ad8456 100755
--- a/caas/mixins.spec
+++ b/caas/mixins.spec
@@ -55,7 +55,7 @@ debug-usb-config: true(source_dev=dvcith-0-msc0)
 intel_prop: true
 tee: trusty(ref_target=celadon_64)
 memtrack: true
-tpm: true
+tpm: false
 avx: auto
 health: hal
 slot-ab: true
-- 
2.34.1

