From 7ee4698b3104510b9b57fff70738eb29d9e4fb95 Mon Sep 17 00:00:00 2001
From: Qi Yadong <yadong.qi@intel.com>
Date: Wed, 11 Dec 2019 09:35:26 +0800
Subject: [PATCH] [DO-NOT-MERGE] enable target debugging for user build
 (console)

Signed-off-by: Qi Yadong <yadong.qi@intel.com>
---
 core/main.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/main.mk b/core/main.mk
index df5c13c..9ea88b4 100644
--- a/core/main.mk
+++ b/core/main.mk
@@ -293,7 +293,7 @@ ifneq (,$(user_variant))
     tags_to_install += debug
   else
     # Disable debugging in plain user builds.
-    enable_target_debugging :=
+    enable_target_debugging := true
   endif
 
   # Disallow mock locations by default for user builds
-- 
2.7.4

