From d89cbcc55b82ed13f12fc3e5bb318440f9709250 Mon Sep 17 00:00:00 2001
From: "ji, zhenlong z" <zhenlong.z.ji@intel.com>
Date: Tue, 16 Nov 2021 10:41:02 +0800
Subject: [PATCH] Print deapexer info

Signed-off-by: ji, zhenlong z <zhenlong.z.ji@intel.com>
---
 tools/releasetools/apex_utils.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/tools/releasetools/apex_utils.py b/tools/releasetools/apex_utils.py
index ef4c69cb2e..9fbffc429c 100644
--- a/tools/releasetools/apex_utils.py
+++ b/tools/releasetools/apex_utils.py
@@ -86,8 +86,10 @@ class ApexApkSigner(object):
     # No need to sign and repack, return the original apex path.
     if not apk_entries:
       logger.info('No apk file to sign in %s', self.apex_path)
+      print("### no apk file to sign in %s\n" % (self.apex_path, ))
       return self.apex_path
-
+    
+    print('Find apk in this apex')
     for entry in apk_entries:
       apk_name = os.path.basename(entry)
       if apk_name not in apk_keys:
-- 
2.25.1

