From d372118eb7eeeefa04f26f848582aaf38e241bfd Mon Sep 17 00:00:00 2001
From: sgnanase <sundar.gnanasekaran@intel.com>
Date: Sun, 29 Mar 2020 22:59:04 +0530
Subject: [PATCH] Fix compilation issue

Signed-off-by: sgnanase <sundar.gnanasekaran@intel.com>
---
 Android.mk | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/Android.mk b/Android.mk
index 59a41927e..7d92e4f35 100644
--- a/Android.mk
+++ b/Android.mk
@@ -253,11 +253,11 @@ LOCAL_REQUIRED_MODULES += \
     odm_mac_permissions.xml
 endif
 
-ifneq ($(PLATFORM_SEPOLICY_VERSION),$(TOT_SEPOLICY_VERSION))
-LOCAL_REQUIRED_MODULES += \
-    sepolicy_freeze_test \
+##ifneq ($(PLATFORM_SEPOLICY_VERSION),$(TOT_SEPOLICY_VERSION))
+#LOCAL_REQUIRED_MODULES += \
+#    sepolicy_freeze_test \
 
-endif # ($(PLATFORM_SEPOLICY_VERSION),$(TOT_SEPOLICY_VERSION))
+#endif # ($(PLATFORM_SEPOLICY_VERSION),$(TOT_SEPOLICY_VERSION))
 
 include $(BUILD_PHONY_PACKAGE)
 
-- 
2.17.1

