From bcb4a1c9dc2c86e095ba6d13bdad3a91da434027 Mon Sep 17 00:00:00 2001
From: Salih <muhammadhx.salih@intel.com>
Date: Thu, 21 Nov 2019 22:28:09 +0530
Subject: [PATCH] Vold: SD and USB coexist-handling

Tracked-On: OAM-91130
Signed-off-by: Salih <muhammadhx.salih@intel.com>
Signed-off-by: N,Shyjumon <shyjumon.n@intel.com>
---
 VolumeManager.cpp | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/VolumeManager.cpp b/VolumeManager.cpp
index 44bff5a..848edf5 100644
--- a/VolumeManager.cpp
+++ b/VolumeManager.cpp
@@ -97,6 +97,7 @@ static const unsigned int kSizeVirtualDisk = 536870912;
 static const unsigned int kMajorBlockMmc = 179;
 static const unsigned int kMajorBlockExperimentalMin = 240;
 static const unsigned int kMajorBlockExperimentalMax = 254;
+static const unsigned int kMajorBlockMmcCIV = 8;	//FIXME: Hack for CIV in IA Arch with KVM
 
 VolumeManager* VolumeManager::sInstance = NULL;
 
@@ -223,7 +224,12 @@ void VolumeManager::handleBlockEvent(NetlinkEvent* evt) {
                                                     major <= (int)kMajorBlockExperimentalMax)) {
                         flags |= android::vold::Disk::Flags::kSd;
                     } else {
-                        flags |= android::vold::Disk::Flags::kUsb;
+			// FIXME: In CIV KVM SD card is virtualized through hdd now,
+			// kMajorBlockMmcCIV and below block need to be  removed.
+			if (major == kMajorBlockMmcCIV && source->getNickname() == "sdcard1")
+				flags |= android::vold::Disk::Flags::kSd;
+			else
+				flags |= android::vold::Disk::Flags::kUsb;
                     }
 
                     auto disk =
-- 
2.24.1

