From aa0a703355732a8e8bef7ba75381c5794c12dfa9 Mon Sep 17 00:00:00 2001
From: pmandri <padmashree.mandri@intel.com>
Date: Thu, 18 Nov 2021 16:34:46 +0530
Subject: [PATCH] Disbale Sound Effects

Tracked-On:
Signed-off-by: pmandri <padmashree.mandri@intel.com>
---
 .../notification/TouchSoundPreferenceControllerTest.java    | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/tests/robotests/src/com/android/settings/notification/TouchSoundPreferenceControllerTest.java b/tests/robotests/src/com/android/settings/notification/TouchSoundPreferenceControllerTest.java
index 91c2a7803b..0410f53fbb 100644
--- a/tests/robotests/src/com/android/settings/notification/TouchSoundPreferenceControllerTest.java
+++ b/tests/robotests/src/com/android/settings/notification/TouchSoundPreferenceControllerTest.java
@@ -84,7 +84,7 @@ public class TouchSoundPreferenceControllerTest {
 
     @Test
     public void displayPreference_soundEffectEnabled_shouldCheckedPreference() {
-        System.putInt(mContentResolver, System.SOUND_EFFECTS_ENABLED, 1);
+        System.putInt(mContentResolver, System.SOUND_EFFECTS_ENABLED, 0);
 
         mController.displayPreference(mScreen);
 
@@ -106,7 +106,7 @@ public class TouchSoundPreferenceControllerTest {
 
         mPreference.getOnPreferenceChangeListener().onPreferenceChange(mPreference, true);
 
-        assertThat(System.getInt(mContentResolver, System.SOUND_EFFECTS_ENABLED, 1)).isEqualTo(1);
+        assertThat(System.getInt(mContentResolver, System.SOUND_EFFECTS_ENABLED, 0)).isEqualTo(1);
     }
 
     @Test
@@ -115,6 +115,6 @@ public class TouchSoundPreferenceControllerTest {
 
         mPreference.getOnPreferenceChangeListener().onPreferenceChange(mPreference, false);
 
-        assertThat(System.getInt(mContentResolver, System.SOUND_EFFECTS_ENABLED, 1)).isEqualTo(0);
+        assertThat(System.getInt(mContentResolver, System.SOUND_EFFECTS_ENABLED, 0)).isEqualTo(0);
     }
 }
-- 
2.17.1

