From 358cfeadfb0cf2ef3d9fb39483630c083575fdb4 Mon Sep 17 00:00:00 2001
From: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
Date: Tue, 11 Dec 2018 15:06:53 +0530
Subject: [PATCH] Changes to make treble complaint

Tracked-On:
Signed-off-by: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
---
 Android.mk | 4 +++-
 lights.c   | 3 ++-
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/Android.mk b/Android.mk
index acfb380..ce4eddc 100644
--- a/Android.mk
+++ b/Android.mk
@@ -22,7 +22,9 @@ LOCAL_SRC_FILES := lights.c
 LOCAL_PRELINK_MODULE := false
 LOCAL_MODULE_RELATIVE_PATH := hw
 
-LOCAL_SHARED_LIBRARIES := liblog
+LOCAL_SHARED_LIBRARIES := liblog libcutils
+LOCAL_HEADER_LIBRARIES := libutils_headers
+LOCAL_HEADER_LIBRARIES += libhardware_headers
 
 LOCAL_MODULE := lights.project-celadon
 
diff --git a/lights.c b/lights.c
index 324ff77..16c30ae 100644
--- a/lights.c
+++ b/lights.c
@@ -16,7 +16,8 @@
 
 #define LOG_TAG "lights"
 
-#include <cutils/log.h>
+#include <unistd.h>
+#include <log/log.h>
 #include <cutils/properties.h>
 #include <stdlib.h>
 #include <string.h>
-- 
1.9.1

