From 1283d870d003bbdb26ddd9a75c25797f322341ad Mon Sep 17 00:00:00 2001
From: "ji, zhenlong z" <zhenlong.z.ji@intel.com>
Date: Mon, 13 Dec 2021 10:08:15 +0800
Subject: [PATCH] Disable the ratelimit for print

Signed-off-by: ji, zhenlong z <zhenlong.z.ji@intel.com>
---
 include/linux/ratelimit.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/linux/ratelimit.h b/include/linux/ratelimit.h
index 8ddf79e9207a..223896a8a93b 100644
--- a/include/linux/ratelimit.h
+++ b/include/linux/ratelimit.h
@@ -6,8 +6,8 @@
 #include <linux/sched.h>
 #include <linux/spinlock.h>
 
-#define DEFAULT_RATELIMIT_INTERVAL	(5 * HZ)
-#define DEFAULT_RATELIMIT_BURST		10
+#define DEFAULT_RATELIMIT_INTERVAL	(0 * HZ)
+#define DEFAULT_RATELIMIT_BURST		1000000
 
 /* issue num suppressed message on exit */
 #define RATELIMIT_MSG_ON_RELEASE	BIT(0)
-- 
2.25.1

