From ee514243f5b4409a724a4b584088f97968f599f0 Mon Sep 17 00:00:00 2001
From: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
Date: Wed, 16 Dec 2020 11:57:12 +0530
Subject: [PATCH] Disable LTO

Tracked-On:
Signed-off-by: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
---
 arch/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/Kconfig b/arch/Kconfig
index 883317dd7da5..eacc249f3b68 100644
--- a/arch/Kconfig
+++ b/arch/Kconfig
@@ -598,7 +598,7 @@ config LTO_CLANG
 	depends on ARCH_SUPPORTS_LTO_CLANG
 	depends on !KASAN
 	depends on !FTRACE_MCOUNT_RECORD || HAVE_C_RECORDMCOUNT
-	depends on CC_IS_CLANG && CLANG_VERSION >= 100000 && LD_IS_LLD
+	depends on CC_IS_CLANG && CLANG_VERSION >= 120000 && LD_IS_LLD
 	select LTO
 	help
           This option enables Clang's Link Time Optimization (LTO), which allows
-- 
2.17.1

