From ca36d1b79a0768990da77b99f35a23dd13670ad2 Mon Sep 17 00:00:00 2001
From: renchenglei <chenglei.ren@intel.com>
Date: Wed, 10 Mar 2021 02:02:53 +0800
Subject: [PATCH] Add shm support for virtio

Tracked-On:
---
 scripts/start_civ.sh | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/scripts/start_civ.sh b/scripts/start_civ.sh
index f54f1fc6a5fa..178fdbe5453c 100755
--- a/scripts/start_civ.sh
+++ b/scripts/start_civ.sh
@@ -325,6 +325,8 @@ function setup_gvtg() {
 
 function setup_virtio_gpu() {
     GUEST_VGA_DEV="-device virtio-gpu-pci"
+    GUEST_SHM_CMD="-device ivshmem-plain,memdev=ivshmem,bus=pcie.0 \
+            -object memory-backend-file,id=ivshmem,share=on,mem-path=/dev/shm/looking-glass,size=32M"
 
     shift #skip first param: VirtIO
 
@@ -654,6 +656,7 @@ function launch_guest() {
               $GUEST_USB_XHCI_OPT \
               $GUEST_STATIC_OPTION \
               $GUEST_EXTRA_QCMD \
+              $GUEST_SHM_CMD \
     "
 
     echo $EXE_CMD
-- 
2.17.1

