From f8dabfc8026cc858464c6c83b26e1987991d8a3a Mon Sep 17 00:00:00 2001
From: renchenglei <chenglei.ren@intel.com>
Date: Sat, 27 Mar 2021 04:55:11 +0800
Subject: [PATCH] Add spice cmd

Tracked-On:
---
 scripts/start_civ.sh | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/scripts/start_civ.sh b/scripts/start_civ.sh
index 79e15e043022..9128efa1062c 100755
--- a/scripts/start_civ.sh
+++ b/scripts/start_civ.sh
@@ -327,6 +327,7 @@ function setup_virtio_gpu() {
     GUEST_VGA_DEV="-device virtio-gpu-pci"
     GUEST_SHM_CMD="-device ivshmem-plain,memdev=ivshmem,bus=pcie.0 \
             -object memory-backend-file,id=ivshmem,share=on,mem-path=/dev/shm/looking-glass,size=32M"
+    GUEST_SPICE_CMD="-spice port=5900,disable-ticketing"
 
     shift #skip first param: VirtIO
 
@@ -657,6 +658,7 @@ function launch_guest() {
               $GUEST_STATIC_OPTION \
               $GUEST_EXTRA_QCMD \
               $GUEST_SHM_CMD \
+              $GUEST_SPICE_CMD \
     "
 
     echo $EXE_CMD
-- 
2.17.1

