From 85789d769b750179a5de9197db7be6a42af782e7 Mon Sep 17 00:00:00 2001
From: "Wan, Hao" <haox.wan@intel.com>
Date: Mon, 15 Apr 2024 17:49:00 +0000
Subject: [PATCH] Resolve the issue of OMX.Intel.sw_vd.h264(5) not being loaded

Signed-off-by: Wan, Hao <haox.wan@intel.com>
---
 media/libstagefright/omx/OMXStore.cpp | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/media/libstagefright/omx/OMXStore.cpp b/media/libstagefright/omx/OMXStore.cpp
index 4827d9e056..dfb7fca9b6 100644
--- a/media/libstagefright/omx/OMXStore.cpp
+++ b/media/libstagefright/omx/OMXStore.cpp
@@ -30,6 +30,8 @@
 
 #include <sstream>
 
+#define OMX_FILTER_VIDEO_CODEC 0
+
 namespace android {
 
 OMXStore::OMXStore() {
@@ -110,6 +112,7 @@ static int getFirstApiLevel() {
     return android::base::GetIntProperty("ro.product.first_api_level", __ANDROID_API_T__);
 }
 
+#if OMX_FILTER_VIDEO_CODEC
 static bool isTV() {
     static const bool kIsTv = []() {
         std::string characteristics = android::base::GetProperty("ro.build.characteristics", "");
@@ -123,11 +126,10 @@ static bool isTV() {
     }();
     return kIsTv;
 }
+#endif
 
 void OMXStore::addPlugin(OMXPluginBase *plugin) {
     Mutex::Autolock autoLock(mLock);
-
-    bool typeTV = isTV();
     int firstApiLevel = getFirstApiLevel();
 
     OMX_U32 index = 0;
@@ -143,15 +145,17 @@ void OMXStore::addPlugin(OMXPluginBase *plugin) {
         if (err == OMX_ErrorNone) {
             bool skip = false;
             for (String8 role : roles) {
+#if OMX_FILTER_VIDEO_CODEC
                 if (role.find("video_decoder") != -1 || role.find("video_encoder") != -1) {
                     if (firstApiLevel >= __ANDROID_API_T__) {
                         skip = true;
                         break;
-                    } else if (!typeTV && firstApiLevel >= __ANDROID_API_S__) {
+                    } else if (!isTV() && firstApiLevel >= __ANDROID_API_S__) {
                         skip = true;
                         break;
                     }
                 }
+#endif
                 if (role.find("audio_decoder") != -1 || role.find("audio_encoder") != -1) {
                     if (firstApiLevel >= __ANDROID_API_T__) {
                         skip = true;
-- 
2.34.1

