From c7c961df5b54ebedbd2c4f08ade67e131fd2fcd6 Mon Sep 17 00:00:00 2001
From: Qi Zhang <qi1.zhang@intel.com>
Date: Tue, 7 Mar 2023 19:31:35 +0800
Subject: [PATCH] remove unknown kernel cmdline buildvariant.

Signed-off-by: Qi Zhang <qi1.zhang@intel.com>
---
 core/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/Makefile b/core/Makefile
index 2d56edb9ee..ac30452ffc 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -898,7 +898,7 @@ VERITY_KEYID := veritykeyid=id:`openssl x509 -in $(PRODUCT_VERITY_SIGNING_KEY).x
 endif
 endif
 
-INTERNAL_KERNEL_CMDLINE := $(strip $(INTERNAL_KERNEL_CMDLINE) buildvariant=$(TARGET_BUILD_VARIANT) $(VERITY_KEYID))
+INTERNAL_KERNEL_CMDLINE := $(strip $(INTERNAL_KERNEL_CMDLINE) $(VERITY_KEYID))
 
 # kernel cmdline/base/pagesize in boot.
 # - If using GKI, use GENERIC_KERNEL_CMDLINE. Remove kernel base and pagesize because they are
-- 
2.34.1

