From bd935febc81401c4984de8cdf90a33b564ccb918 Mon Sep 17 00:00:00 2001
From: Sapna <sapna1.singh@intel.com>
Date: Thu, 3 Apr 2025 10:40:45 +0530
Subject: [PATCH] Make the headers available for other projects

Signed-off-by: Sapna <sapna1.singh@intel.com>
---
 Android.bp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/Android.bp b/Android.bp
index 5537fb0d..13d33253 100644
--- a/Android.bp
+++ b/Android.bp
@@ -37,6 +37,14 @@ license {
     ],
 }
 
+cc_library_headers {
+    name: "expat_headers",
+
+    export_include_dirs: [
+        "lib",
+    ],
+}
+
 cc_library {
     name: "libexpat",
     vendor_available: true,
-- 
2.42.0

